
## Comprehensive Genome - wide Protein-DNA interactions Detected at Single-Nucleotide Resolution

#### Note

A sketch of the steps followed by the ChIP-Exo protocol is on the notes of  Rhee & Pugh, 2012.

### Motivation

On a ChIP Seq experiment, the unbound DNA contaminates the inmunoprecipitate, ChIP (chip or seq) provides a set of statistically enriched high-occupancy binding regions, rather that a complete and precise set of bound locations. A large fraction of this set of regions could be false positive and many other sites could be missed (false negatives).

Also, from this regions, motif searches (MEME for example) are insufficient to identify all in vivo biding locations for proteins because proteins recognize a wide variety of related sequences, of which only a small fraction are bound, consequently, althrough a consensus motif could be found, a large fraction of putatively found locations could either lack or contain a degenerate version of the motif.

### Comments on results

With ChIP-Exo it is possible to find the motifs found by ChIP-Seq, and also a collection of additional binding sites (check fig S6 - B).

The motifs for yeast are confounded, i.e. are not exactly the same as the motifs found by using ChIP-Seq data. But that motif could be obtained into the motifs find in the ChIP-Exo data all together.

### Discussion

Claim: The false-positive and false-negative rates associated with ChIP-Chip and ChIP-Seq vary depending upon chromatin fragmentation heterogeneity. The author suspect that 50% of factor bound location found by ChIP-Chip and 30% of location found by ChIP-Seq could be false positives, i.e. ChIP-Exo disminishes the overall number of false positives.




[The paper could be found here](http://www.sciencedirect.com/science/article/pii/S0092867411013511)



