
## Alignment summary

Overall the alignment algorithm is:

1. Run fastqc to review some graphical indicators of the quality of the reads
2. Run trimmomatic (to review what is doing exactly)

[trimmomatic documentation can be found here](http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.30.pdf)


3. Run bwa [bwa documentation can be found here](http://bio-bwa.sourceforge.net/bwa.shtml)

 ..1 bwa index - Builds index to align the reads
 ..2 bwa aln / bwa samse - Align using bwa backtrack algorithm

4. Use samtools to convert to bam [samtools documentation can be found here](http://samtools.sourceforge.net/samtools.shtml)

### Command for trimmomatic

```
trimAssemble <- function(leftDirtyName, rightDirtyName=NULL, trimPath, qScore,
                         headcrop=12, artifactsFile, myVersion, libMode, targetQ, minlen) {
  dashqScore <- paste("-", qScore, sep="")
  logName <- paste(leftDirtyName, "pairedtrim.log", sep=".")
  pairedTrimmed.1 <- paste("p", myVersion, ".", leftDirtyName, sep="")
  unpairedTrimmed.1 <- paste("u", myVersion, ".",  leftDirtyName, sep="")
  pairedTrimmed.2 <- paste("p",  myVersion, ".", rightDirtyName, sep="")
  unpairedTrimmed.2 <- paste("u", myVersion, ".", rightDirtyName, sep="") 
  trimParam <- paste("ILLUMINACLIP:", artifactsFile, ":2:30:10", 
                     " HEADCROP:", headcrop,
                     " SLIDINGWINDOW:3:", targetQ,
                     " MINLEN:", minlen, sep="")
  if (libMode == "PE")
    tcomm <- paste("java -jar", trimPath, "PE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   rightDirtyName, pairedTrimmed.1, unpairedTrimmed.1, 
                   pairedTrimmed.2, unpairedTrimmed.2, trimParam)
  if (libMode == "SE")
    tcomm <- paste("java -jar", trimPath, "SE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   pairedTrimmed.1, trimParam)
tcomm }
```

In general trimmomatic is applied to one of the following two cases: 
- Paired end tags (PET)
- Single end tags (SET) 

This comes from the trimmomatic manual, is the quick reference of the implemented trimming steps used:
- ILLUMINACLIP:<artifactsFile>:2:30:10 - Cut adapter and other Illumina-specific sequences from the read 
  - It accepts a maximum number of 2 mismatches
  - 30 bp match for between two 'adapter ligated' reads must be for PE palindrome read alignment
  - 10 bp match between any 'adapter ligated' sequence must be against a read

- HEADCROP - Cut the specified number of bases from the start of the read
- SLIDINGWINDOW:<window size>:<required quality> - Perform a sliding window approach. It starts scanning at the 5' end and clips the read once the average quality within the window falls bellow a threshold
- MINLEN: Drop the read if its below a specified length
- 

### for ChIPexo or TSS
# headcrop=0, targetQ=0, minlen=0 -- (this will leave the reads as is)
# headcrop=0, targetQ=26, minlen=25 -- (This way, you'll remove the low-quality stuff from the 3'-end and will have the 5'-end intact)








