
## Alignment summary

Overall the alignment algorithm is:

1. Run fastqc to review some graphical indicators of the quality of the reads
2. Run trimmomatic (to review what is doing exactly)
3. Run bwa

 ..1 bwa index - Builds index to align the reads
 ..2 bwa aln / bwa samse - Align using bwa backtrack algorithm

4. Use samtools to convert to bam


### Command for trimmomatic

```
trimAssemble <- function(leftDirtyName, rightDirtyName=NULL, trimPath, qScore,
                         headcrop=12, artifactsFile, myVersion, libMode, targetQ, minlen) {
  dashqScore <- paste("-", qScore, sep="")
  logName <- paste(leftDirtyName, "pairedtrim.log", sep=".")
  pairedTrimmed.1 <- paste("p", myVersion, ".", leftDirtyName, sep="")
  unpairedTrimmed.1 <- paste("u", myVersion, ".",  leftDirtyName, sep="")
  pairedTrimmed.2 <- paste("p",  myVersion, ".", rightDirtyName, sep="")
  unpairedTrimmed.2 <- paste("u", myVersion, ".", rightDirtyName, sep="") 
  trimParam <- paste("ILLUMINACLIP:", artifactsFile, ":2:30:10", 
                     " HEADCROP:", headcrop,
                     " SLIDINGWINDOW:3:", targetQ,
                     " MINLEN:", minlen, sep="")
  if (libMode == "PE")
    tcomm <- paste("java -jar", trimPath, "PE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   rightDirtyName, pairedTrimmed.1, unpairedTrimmed.1, 
                   pairedTrimmed.2, unpairedTrimmed.2, trimParam)
  if (libMode == "SE")
    tcomm <- paste("java -jar", trimPath, "SE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   pairedTrimmed.1, trimParam)
tcomm }
```


