
## Alignment summary

Overall the alignment algorithm is:

1. Run fastqc to review some graphical indicators of the quality of the reads
2. Run trimmomatic (to review what is doing exactly)

[trimmomatic documentation can be found here](http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.30.pdf)


3. Run bwa [bwa documentation can be found here](http://bio-bwa.sourceforge.net/bwa.shtml)

 ..1 bwa index - Builds index to align the reads
 ..2 bwa aln / bwa samse - Align using bwa backtrack algorithm

4. Use samtools to convert to bam [samtools documentation can be found here](http://samtools.sourceforge.net/samtools.shtml)

### Command for trimmomatic

```
trimAssemble <- function(leftDirtyName, rightDirtyName=NULL, trimPath, qScore,
                         headcrop=12, artifactsFile, myVersion, libMode, targetQ, minlen) {
  dashqScore <- paste("-", qScore, sep="")
  logName <- paste(leftDirtyName, "pairedtrim.log", sep=".")
  pairedTrimmed.1 <- paste("p", myVersion, ".", leftDirtyName, sep="")
  unpairedTrimmed.1 <- paste("u", myVersion, ".",  leftDirtyName, sep="")
  pairedTrimmed.2 <- paste("p",  myVersion, ".", rightDirtyName, sep="")
  unpairedTrimmed.2 <- paste("u", myVersion, ".", rightDirtyName, sep="") 
  trimParam <- paste("ILLUMINACLIP:", artifactsFile, ":2:30:10", 
                     " HEADCROP:", headcrop,
                     " SLIDINGWINDOW:3:", targetQ,
                     " MINLEN:", minlen, sep="")
  if (libMode == "PE")
    tcomm <- paste("java -jar", trimPath, "PE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   rightDirtyName, pairedTrimmed.1, unpairedTrimmed.1, 
                   pairedTrimmed.2, unpairedTrimmed.2, trimParam)
  if (libMode == "SE")
    tcomm <- paste("java -jar", trimPath, "SE -threads 20", dashqScore,
                    "-trimlog", logName, leftDirtyName,
                   pairedTrimmed.1, trimParam)
tcomm }
```

In general trimmomatic is applied to one of the following two cases: 
- Paired end tags (PET)
- Single end tags (SET) 

This comes from the trimmomatic manual, is the quick reference of the implemented trimming steps used:
- ILLUMINACLIP:<artifactsFile>:2:30:10 - Cut adapter and other Illumina-specific sequences from the read 
  - It accepts a maximum number of 2 mismatches
  - 30 bp match for between two 'adapter ligated' reads must be for PE palindrome read alignment
  - 10 bp match between any 'adapter ligated' sequence must be against a read

- HEADCROP:<length> - Cut the specified number of bases from the start of the read
  - The number of bases to remove of the start of each read. In this case is always zero

- SLIDINGWINDOW:<window size>:<required quality> - Perform a sliding window approach. It starts scanning at the 5' end and clips the read once the average quality within the window falls bellow a threshold
  - The window size applied to all experiments is 3 

- MINLEN<length>: Drop the read if its below a specified length 
  - Specifies the minimum length of the reads to be kept

**Brief remark for ChIP exo or TSS**
+ headcrop=0, targetQ=0, minlen=0 -- (this will leave the reads as is)
+ headcrop=0, targetQ=26, minlen=25 -- (This way, you'll remove the low-quality stuff from the 3'-end and will have the 5'-end intact)

### Summary of alignment files

#### ChIP Exo:

Columns:
- edns - Experimental data set number ---- this are unique
- cult - Cultivation number 

|  edns |  cult  |  Rif time  |  IP |  Rep  |  Surviving | Dropped |
| ---   | ---    | ---        | --- | ---   | ---        | ---     |
| 1310  | 1197   | 0 min      | Beta | 1      |  5,342,809 (94.63%) | 302,971 (5.37%) |
| 1311  | 1197   | 0 min      | Sig70| 1      | 1,002,850 (95.45%) | 47,830 (4.55%)|
| 1312  | 1197   | 0 min      | BetaPrimeFlag | 1 | 1,940,436 (94.98%) | 102,665 (5.02%) |
| 1313  | 1197   | 20 min     | Beta | 1      | 19,296,033 (93.02%) | 1,446,921 (6.98%) |
| 1314  | 1197   | 20 min     | Sig70| 1      | 2,023,464 (95.63%)   | 92,394 (4.37%)  |
| 1315  | 1197   | 20 min     | BetaPrimeFlag | 1|6,528,691 (95.09%)  | 336,894 (4.91%) |
| 1316  | 1202   | 0 min      | Beta | 2      |  5,923,547 (94.03%) | 376,279 (5.97%) |
| 1317  | 1202   | 0 min      | Sig70| 2      | 2,345,504 (95.93%) | 99,394 (4.07%)|
| 1318  | 1202   | 0 min      | BetaPrimeFlag | 2 | 925,859 (95.30%) | 45,657 (4.70%) |
| 1319  | 1202   | 20 min     | Beta | 2      | 16,017,071 (83.79%) | 3,098,183 (16.21%) |
| 1320  | 1202   | 20 min     | Sig70| 2      | 12,604,747 (95.26%)   |  626,954 (4.74%) |
| 1321  | 1202   | 20 min     | BetaPrimeFlag |2 |  22,332,347 (96.00%) | 929,638 (4.00%) |


#### ChIP Seq - PET

|  edns |  cult  |  Rif time  |  IP |  Rep  |  Surviving Both | Fwd only surviving | Rev only surviving | Dropped |
| ---   | ---    | ---        | --- | ---   | ---             | ---                | ---                | ---     |
| 1396  | 1197   | 0 min   | Sig70 | 1   | 5,443,451 (46.03%) | 3,935,239 (33.28%) | 288,341 (2.44%) | 2,158,174 (18.25%) |
