%\VignetteEngine{knitr::knitr}
\documentclass{article}

<<<style-knitr, eval=TRUE, echo=FALSE, results="asis">>=
  BiocStyle::latex()
@ 


\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{float}
\usepackage{graphicx}


\begin{document}

\title{Comments on creating the partition with slice function}

\maketitle

\section{Introduction}

The previous analysis showed that the partition of the e.coli genome
could improve the behavior of the summary statistics considered.

We improved the partition by considering the \Rfunction{coverage} of
both forward and backward reads. Then we calculated the regions by
using the \Rfunction{slice} function considering a lower bound of at
least 5 counts.

We labeled the regions according to the following cases:

\begin{itemize}
\item If the island of both strands overlapped then we labeled as
  ``both'' and considered the union of all overlapping islands as
  regions

\item Otherwise, we labeled as ``fwd'' of ``bwd'' if it belong to
  their respective strand and didn't overlapped a regions of the
  opposite strand

\end{itemize}

Given a region we calculated the following summary statistics:

\begin{enumerate}
\item Forward strand ratio:

  \begin{align}
    \text{ratio} = \frac{f}{f+r}
  \end{align}

  where $f$ and $r$ are the number of forward and reverse reads that
  overlap the region

\item Max. strand summit position:

  \begin{align}
    \text{summit pos}(\mathcal{S}) = \arg\max_{x} \text{cover}(x) 
  \end{align}

  where $\text{cover}$ is the coverage calculated by a set of reads
  $\mathcal{S}$. In the case that there are more than one positions ts
  chooses the one closest to the 5' end.

\item Difference of summit positions:

  \begin{align}
    \text{diff} = \text{summit pos}(\mathcal{S^{-}}) - \text{summit pos}(\mathcal{S^{+}})
  \end{align}

  This is the the difference between the summit position between the
  reverse reads minus the forward reads.

\end{enumerate}

\newpage

\section{Figure \#1}

The first thing, was to analyze the width of the regions obtained by
the partition method:

\begin{figure}[H]
  \centering
  \includegraphics[width = .6\textwidth]{../inst/figs/summaryStats/widthSlice.pdf}
  \caption{Histogram of the with of the regions separated by label}
\label{width}
\end{figure}

In figure \ref{width} we can see:

\begin{itemize}
\item The regions labeled as ``both'' are usually wider, that may be
  of the way they were constructed.

\item The regions labeled as ``fwd'' or ``bwd'' seem to present the
  same behavior.

\item There are very wide regions but that's the less common case in
  all three cases. 

\item For the strand specific cases there are several regions with
  width $\leq 50$ which must correspond to cases where more than five
  reads are being mapped to the same position

\item The mode of the histogram is happening for regions approximately
  70 bp long, which must correspond to cases with very few position
  mapped to very close positions

\end{itemize}

After watching the regions, I think that a possible improvement could
be to apply the procedure to generate the regions to both replicates
(in this case I only did it for the replicate with highest depth) and
perhaps merge the regions, i.e. obtain for each replicate a set of
regions for each strand, then merge for each strand and then label
according to the same idea.

\newpage

\section{Figures \# 2}

In the following plots we are considering a different panel per
replicate and treatment. Then insider each panel we are calculating
the boxplot for the forward strand ratio of each sequencing technology
by which reads were generated (or sampled). For the second figure, we
are colouring the boxplots by dividing them by label (i.e. ``both'',
``fwd'' or ``bwd'')


\begin{figure}[H]
  \centering
  \includegraphics[width=.85\textwidth,page = 1]{../inst/figs/summaryStats/FwdStrandRatioSlice.pdf}
  \caption{Boxplot of forward strand ratio separated by data sets}
\label{ratioBad}
\end{figure}

On figure \ref{ratioBad} we can see that the forward strand ratio is
behaving as expected. Is localized around $0.5$, the range of the
ChIP-seq samples is smalled than the ChIP-exo one.



\begin{figure}[H]
  \centering
  \includegraphics[width=0.85\textwidth,page=2]{../inst/figs/summaryStats/FwdStrandRatioSlice.pdf}
  \caption{Boxplot of forward strand ratio separated by data set and label}
\label{ratioOk}
\end{figure}

In figure \ref{ratioOk} we can see that the more imbalanced regions
(those which correspond to a ratio of 0 or 1) belong to the ``fwd'' or
``bwd'' regions. In particular the regions labeled as ``both'' seem to
localize around 0.5 and the range is less than the half of the range
considered of the statistic considering all labels together (except for $\beta'_{F}$ and rif applied by 0 min).

% do the case for the \beta ip



% Get a table with the number of labeled regions for data set

% [1] 1
% [1] 2315
% [1] 4390
% [1] 4524
% [1] 2
% [1] 5303
% [1] 5811
% [1] 5813
% [1] 3
% [1] 2927
% [1] 5169
% [1] 5063
% [1] 4
% [1] 4303
% [1] 5319
% [1] 5263

\section{Figures \# 3}

In this plot, we are separating the panels by treatment and
replicate. In each panel we are showing the estimated density
function, of the difference between summit positions (backward -
forward, this was to be considering mostly positive numbers).  

For ChIP-seq the label separation looks as following:

\begin{figure}[H]
  \centering 
\includegraphics[width=0.85\textwidth,page=5]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\includegraphics[width=0.85\textwidth,page=6]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\caption{Density of difference in summit positions separated
    by label for ChIP-seq PET (above) and SET(below)}
\label{dens-chipseq}
\end{figure}

From both plots in figure \ref{dens-chipseq}, we can expect the
``fwd'' and ``bwd'' labels to be centered around zero. And for the
``both'' label, we can expect to have a positive value. 

For all labels we can see that there are some negative values but that
is not the most common case. Both plots look very similar but that is
happening because the SET data set was sampled from the PET data set.

\begin{figure}[H]
  \centering 
\includegraphics[width=0.85\textwidth,page=4]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\caption{Density of difference in summit positions separated
    by label for ChIP-exo}
\label{dens-exo}
\end{figure}

For ChIP-exo, we are expecting to see the mode (or modes) of the
distribution to have a small positive values. Also, we expect it to
have only non-negative values (which is not the case, however this
plot considers all the regions and not only the ones where there is a
binding event). A promising feature is that the density for the
``both'' label seem to be lower than the densities of the other
labels. Perhaps by considering peaks, we can trim this density.

Finally we can see figure \ref{dens-both} which show the density of all the sequencing procedures for the ``both'' label:

\begin{figure}[H]
  \centering 
\includegraphics[width=0.85\textwidth,page=1]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\caption{Density of difference in summit positions separated
    by sequencing for ``both'' label}
\label{dens-both}
\end{figure}

Figure \ref{dens-both} looks very promising since, by considering a
very simple classification we are seeing that the difference for the
ChIP-exo data set is lower than the difference for the ChIP-seq cases. There are a large amount of cases where the difference is negative, which may be filtered by considering them in an case by case basis.

% perhaps considering the regions where the depth of the region is high and are labeled as both

\begin{figure}[H]
  \centering 
\includegraphics[width=0.85\textwidth,page=2]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\includegraphics[width=0.85\textwidth,page=3]{../inst/figs/summaryStats/SummitPosDiffSlice.pdf}
\caption{Density of difference in summit positions separated
    by sequencing for ``fwd'' and ``bwd'' labels}
\label{dens-both}
\end{figure}



\end{document}
