%\VignetteEngine{knitr::knitr}
\documentclass{article}

<<<style-knitr, eval=TRUE, echo=FALSE, results="asis">>=
  BiocStyle::latex()
@ 

\usepackage{amsmath}
\usepackage{amssymb}
\usepackage{float}
\usepackage{graphicx}


\begin{document}

\title{An analysis of quality control measures for Chip-exo}
\author{Rene Welch \\ Department of Statistics, University of Wisconsin-Madison \\ Madison, WI}
\date{October 2014}

\maketitle


\tableofcontents

\section{Introduction}


The idea of this document is to summarize the different analysis made
to asses the quality of different ChIP-exo data sets. Several analysis
were performed to the ChIP exo data.

\subsection{The data}



\section{PCR bottleneck coefficient}

The PCR bottleneck coefficients is defined as a measure of library
complexity, i.e. how skewed the distribution of read counts per
location is towards 1 read per location. It is defined as:

\begin{align} 
  \text{PBC} = \frac{N_1}{N_d}
\end{align}

where:

\begin{itemize}

\item $N_1$ is the number of genomic locations to which
  \textbf{exactly} one unique mapping read maps

\item $N_d$ is the number of genomic locations to which \textbf{at
    least} one unique mapping read maps, i.e. the number of
  non-redundant unique mapping reads

\end{itemize}

Since $N_1 \leq N_d$ then $0 \leq \text{PBC} \leq 1$. Finally ENCODE
recommends:

\begin{table}[H]
  \centering
  \begin{tabular}{c|c}
    \hline\hline
    PBC range & Bottleneck class \\
    \hline
    0 - 0.5  & Severe  \\
    0.5 - 0.8 & Moderate  \\
    0.8 - 0.9 & Mild        \\  
    0.9 - 1 & Non-existent  \\
    \hline
  \end{tabular}
  \caption{PBC classification \label{pbc_class}}
\end{table}

We grouped the data sets by the protocol it was used to generate
it. We considered three protocols: ChIP-exo, ChIP-seq PET and ChIP-seq
SET. We wanted to compare if there is an overall level across the same
protocols used. Thus, for each data set, we calculated it's PBC and
plotted all together by grouping them across protocols:

\begin{figure}[H]
  \centering
  \includegraphics[width =
  .4\textwidth]{../inst/figs/pbc/PBCboxplot.pdf}
  \caption{PCR bottleneck coefficient separated by
    protocol \label{pbcfig}}
\end{figure}

In figure \ref{pbcfig} we can see that PBC is higher for both ChIP-seq
protocols. Also, we can see that the range of ChIP-Seq SET PBC is
bigger than the other two.

\section{Cross-correlation}

A problem for ChIP-seq SET data sets is that the read fragment length
is unknown. Cross-correlation is a method that gives an estimation of
this quantity by the shift where the strand cross - correlation is
maximized. For a given shift length $\delta$, the strand
cross-correlation is defined as:

\begin{align}
  \rho(\delta) = \sum_{c\in C} \frac{N_c}{N} P\left[n_c^+(x + \delta
    /2), n_c^-(x - \delta / 2) \right]
  \label{cross-corr}
\end{align}

where:

\begin{itemize}
\item $N_c$ is the number of reads mapped to chromosome $c$ and $N$ is
  the sum of all the reads mapped to the genome
\item $P(\bf{x},\bf{y})$ is the Pearson's correlation between vectors
  $\bf{x}$ and $\bf{y}$
\item $n_c^S(x)$ is the tag count vector of chromosome $c$ and strand
  $S$, centered at position $x$
\end{itemize}

For this part of the analysis we calculated the cross - correlation curve using two different pipelines: \Rpackage{spp} and \Biocpkg{ChIPQC} and compared both curves visually.

<<load_cross_data,include=FALSE,echo=FALSE,eval=TRUE>>=
  library(ggplot2)
  datadir = "../data"
  load(file.path(datadir,"cross_corr.RData"))
@  

<<chipseqSET,include=FALSE,echo=FALSE,eval=TRUE>>=
  df = set_df_list[[3]]       
  ggplot(df,aes(shift,crossCorr,colour = method))+geom_line()+theme(legend.position = "bottom",aspect.ratio = 1/3,axis.title = element_text(size = rel(2.5)),axis.text = element_text(size = rel(2)),legend.text = element_text(size = rel(2)),legend.title = element_text(size = rel(2)))+ scale_color_brewer(palette ="Dark2")
  ggsave("setCC1.pdf",width = 12,height = 5)
@ 

<<chipseqPET,include=FALSE,echo=FALSE,eval=TRUE>>=
  df = pet_df_list[[5]]       
  ggplot(df,aes(shift,crossCorr,colour = method))+geom_line()+theme(legend.position = "bottom",aspect.ratio = 1/3,axis.title = element_text(size = rel(2.5)),axis.text = element_text(size = rel(2)),legend.text = element_text(size = rel(2)),legend.title = element_text(size = rel(2)))+ scale_color_brewer(palette ="Dark2")
  ggsave("setCC2.pdf",width = 12,height = 5)
@ 

\begin{figure}[H]
  \centering
  \includegraphics[width=.8\textwidth]{setCC1.pdf}
  \caption{Strand cross-correlation for a typical ChIP-seq SET data set}
  \label{fig:cc1}
\end{figure}

\begin{figure}[H]
  \centering
  \includegraphics[width=.8\textwidth]{setCC2.pdf}
  \caption{Strand cross-correlation for a typical ChIP-seq PET data set}
  \label{fig:cc2}
\end{figure}

In both figures \ref{fig:cc1} and \ref{fig:cc2} we can see typical cross correlation curves for both ChIP-seq SET and PET data sets. In this case, there are no phantom peaks or any weird effect in the curves. So, when estimating the fragment length using the shift where each curve is maximized we can see that the position doesn't variate a lot between methods (it may be a matter of some bp). Also, it's is worth noticing that the range of both curves is on a decimal scale. However, when we plot one of the ChIP-exo datasets, we can see that its curves are both in a centesimal scale:

<<chipseqEXO,include=FALSE,echo=FALSE,eval=TRUE>>=
  df = exo_df_list[[8]]       
  ggplot(df,aes(shift,crossCorr,colour = method))+geom_line()+theme(legend.position = "bottom",aspect.ratio = 1/3,axis.title = element_text(size = rel(2.5)),axis.text = element_text(size = rel(2)),legend.text = element_text(size = rel(2)),legend.title = element_text(size = rel(2)))+ scale_color_brewer(palette ="Dark2")
  ggsave("setCC3.pdf",width = 12,height = 5)
@ 

\begin{figure}[H]
  \centering
  \includegraphics[width=.8\textwidth]{setCC3.pdf}
  \caption{Strand cross-correlation for a typical ChIP-exo data set}
  \label{fig:cc3}
\end{figure}

That means that both strand are approximately uncorrelated. Thus, this method may not work to estimate the read fragment length for a ChIP-exo data set. Also, we can think that this low values may be an effect of having sparse tag count vectors (which may be happening because of the enzyme diggesting all positions from the 5' end until reaching the TF.




\section{Forward strand ratio density}




\subsection{Forward strand ratio density estimation conditional to higher count bins}


\section{Examples}



\end{document}
