%\VignetteEngine{knitr::knitr}
\documentclass{bmcart}

\usepackage{xcolor}
\usepackage{url}
\usepackage{amsmath}
\usepackage{amsthm}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{tikz}
\usetikzlibrary{shapes,arrows}
\usepackage{float}
\usepackage{verbatim}

% \usepackage{authblk}
\usepackage[utf8]{inputenc} %unicode support


\begin{document}

%\maketitle

\begin{frontmatter}

\begin{fmbox}
\dochead{Draft}

\title{ChIP-exo: High Resolution Identification of Protein-DNA Binding
  Events and Quality Control}

\author[
   addressref={aff6},                   % id's of addresses, e.g. {aff1,aff2}
   email={chungd@musc.edu}   % email address
]{\inits{DC}\fnm{Dongjun} \snm{Chung}}
\author[
   addressref={aff1},
   email={welch@stat.wisc.edu}
]{\inits{RW}\fnm{Rene} \snm{Welch}}
\author[
   addressref={aff3},
   email={ong@cs.wisc.edu}
]{\inits{IO}\fnm{Irene} \snm{Ong}}
\author[
   addressref={aff3,aff4},
   email={jagrass@wisc.edu}
]{\inits{JG}\fnm{Jeffrey} \snm{Grass}}
\author[
   addressref={aff3,aff4,aff5},
   email={landick@bact.wisc.edu}
]{\inits{RL}\fnm{Robert} \snm{Landick}}
\author[
   addressref={aff1,aff2},
   corref={aff1},
   email={keles@stat.wisc.edu}
]{\inits{SK}\fnm{S\"und\"uz} \snm{Kele\c{s}}}



\address[id=aff1]{%                           % unique id
  \orgname{Department of Statistics, University of Wisconsin Madison}, % university, etc
  \street{1300 University Avenue},                     %
  %\postcode{}                                % post or zip code
  \city{Madison},                              % city
  \cny{WI}                                    % country
}

\address[id=aff2]{%
  \orgname{Department of Biostatistics and Medical Informatics, University of Wisconsin Madison},
  \street{600 Highland Avenue},
%  \postcode{24105}
  \city{Madison},
  \cny{WI}
}
\address[id=aff3]{
  \orgname{Great Lakes Bioenergy Research Center, University of Wisconsin Madison},
  \street{1552 University Avenue},
  \city{Madison},
  \cny{WI}
}
\address[id=aff4]{
  \orgname{Department of Biochemistry, University of Wisconsin Madison},
  \street{433 Babcock Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff5]{
  \orgname{Department of Bacteriology, University of Wisconsin Madison},
  \street{1550 Linden Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff6]{
  \orgname{Department of Public Health Sciences, Medical University of South Carolina},
  \street{135 Cannon Street},
  \city{Charleston},
  \cny{SC}
}


% \begin{artnotes}
% %\note{Sample of title note}     % note to the article
% \note[id=n1]{Equal contributor} % note, connected to author
% \end{artnotes}


\end{fmbox}

\begin{abstractbox}

  \begin{abstract}

  Recently, ChIP-exo has been developed to investigate protein-DNA
  interaction in higher resolution compared to popularly used
  ChIP-Seq. Although ChIP-exo has drawn much attention and is
  considered as powerful assay, currently, no systematic studies have
  yet been conducted to determine optimal strategies for experimental
  design and analysis of ChIP-exo. In order to address these
  questions, we evaluated diverse aspects of ChIP-exo and found the
  following characteristics of ChIP-exo data. First, Background of
  ChIP-exo data is qute different from that of ChIP-Seq data. However,
  sequence biases inherently present in ChIP-Seq data still exist in
  ChIP-exo data. Second, in ChIP-exo data, reads are located around
  binding sites much more tightly and hence, it has potential for high
  resolution identification of protein-DNA interaction sites, hence
  the space to allocate the reads is greatly reduced. Third, although
  often assumed in the ChIP-exo data analysis methods, the peak pair
  assumption does not hold well in real ChIP-exo data. Fourth, spatial
  resolution of ChIP-exo is comparable to that of PET ChIP-Seq and
  both of them are significantly better than resolution of SET
  ChIP-Seq. Finally, for given fixed sequencing depth, ChIP-exo
  provides higher sensitivity, specificity, and spatial resolution
  than PET ChIP-Seq.

  In this article, we provide a quality control pipeline which
  visually asses ChIP-exo biases and calculates a signal-to-noise
  measure. Also, we updated dPeak \cite{dpeak}, which makes a striking
  balance in sensitivity, specificity, and spatial resolution for
  ChIP-exo data analysis.

\end{abstract}
\begin{keyword}
  \kwd{ChIP-exo}
  \kwd{QC}
  \kwd{TFBS}
  \kwd{BS identification on High-Res}
\end{keyword}

\end{abstractbox}

\end{frontmatter}

\newpage

* to remove later

\tableofcontents


\newpage


\section{Background}
\label{sec:intro}

ChIP-exo (Chromatin Immunopecipitation followed by exonuclease
digestion and next generation sequencing) Rhee and Pugh (\cite{exo1})
is the state-of-the-art experiment developed to attain single
base-pair resolution of protein binding site identification and it is
considered as a powerful alternative to popularly used ChIP-Seq
(Chromatin Immunoprecipitation coupled with next generation sequencing
) assay. ChIP-exo experiments first capture millions of DNA fragments
(150 - 250 bp in length) that the protein under study interacts with
using random fragmentation of DNA and a protein-specific
antibody. Then, exonuclease is introducted to trim 5' end of each DNA
fragment to a fixed distance from the bound protein. As a result,
boundaries around the protein of interest constructed with 5' ends of
fragments are located much closed to bound protein compared to
ChIP-Seq. This is the step unique to ChIP-exo that could potentially
provide significantly higher spatial resolution compared to
ChIP-Seq. Finally, high throughput sequencing of a small region (25 to
100 bp) at 5' end of each fragment generates millions of reads or
tags.

While the number of produced ChIP-exo data keeps increasing,
characteristicos of ChIP-exo data and optimal strategies for
experimental design and analysis of ChIP-exo data are not fully
investigated yet, including issues of sequence biases inherent to
ChIP-exo data, choice of optimal statistical methods, and
determination of optimal sequencing deoth. However, currently, the
number of available ChIP-exo data is still limited and their
sequencing depths are still insufficient for such investigation. To
address this limitation we gathered ChIP exo data from diverse
organisms: CTCF factor in human \cite{exo1}; ER factor in human and
FoxA1 factor in mouse from \cite{exoillumina}; and generated
$\sigma^{70}$ factor in Escherichia coli (E. Coli) under aerobic ($ +
O_2$) condition, and treated by rifampicin by 0 and 20 minutes.

DNA libraries generated by the ChIP-exo protocol seem to be less
complex than the libraries generated by ChIP-Seq
\cite{exo_review}. Hence, most of current guidelines \cite{encode_qc}
may not be aplicable on ChIP-exo. To address this challenge, we
suggest a series of quality control visualizations to understand which
biases are present in ChIP-exo data. Related to quality control,
Previous ChIP-exo analysis used ChIP-Seq samples to compare the
resolution between experiments (\cite{exo1}, \cite{exo2},
\cite{exoillumina}). In \cite{carroll.qc}, Carroll et. al. studied the
use of the Strand-Cross Correlation (SCC) \cite{strandcc}. and showed
that by filtering blacklisted regions the estimation of the SCC is
improved. However, this method requires to know blacklisted regions in
advance which may not be available. In our pipeline we propose two
out-the-shelf metrics equivalent to RSC and NSC for the estimation of
the signal-to-noise ratio for a ChIP-exo sample.

In order to archieve the potential benefits of ChIP-exo on protein
binding site identification, it is critical to understand which are
the important characteristics of ChIP exo data and to use algorithms
that could fully utilize information available in ChIP exo data. Rhee
and Pugh \cite{exo1} discussed that reads in the forward and reverse
strand might construct peak pairs around bound protein, of which
heights were implicitly assumed to be symmetric. Hence, they used the
``peak pair method'' that predicts the midpoint of two modes of peak
pairs as potential binding site. Mace \cite{mace}, CexoR \cite{cexor}
and peakzilla \cite{peakzilla}, recently developed ChIP-exo data
analysis methods, are also based on this peak pair
assumption. However, appropriatness of such assumption was not fully
evaluated in the literature yet.  Furthermore, it is still unknown
which factors could affect protein binding site ientification using
ChIP exo data. In order to address this problem, we investigated
various aspects of ChIP exo data by contrasting them with their
respective ChIP Seq experiments.

Currently, research on statistical methods for ChIP-exo data is still
in its very early stage. Although many methods have been proposed to
identify protein binding sites from ChIP Seq data (reviewed in
\cite{evaluation} and \cite{computation}), such as MACS \cite{macs},
CisGenome \cite{cisgenome} and MOSAiCS \cite{mosaics}, these
approaches reveal protein binding sites in lower resolution, i.e., at
an interval of hundreds to thousands of base pairs. Furthermore, they
report only one ``mode'' or ``predicted binding location'' per
peak. Hence, these methods are not appropiate to evaluate the
potential of ChIP-exo data for high resolution identification of
protein binding sites. More recently, deconvolution algorithms such as
CSDeconv \cite{csdeconv}, GEM \cite{gem} (an improved version of
\cite{gps}) and PICS \cite{pics} have been proposed to identify
binding sites in higher resolution using ChIP-Seq data. However, most
of them are still not tailored for ChIP-exo and PET and SET ChIP-Seq
data in a unified framework and as a result, currently available
methods are not appropiate for fair comparison between ChIP-exo and
ChIP-Seq. To address these limitations, we developed an improved of
dPeak \cite{dpeak}, a high resolution binding site identification
(deconvolution) algorithm that we previously developed for PET and SET
ChIP-Seq data, so that it can also handle ChIP-exo data. The dPeak
algorithm implements a probabilistic model that accurately describes
the ChIP-exo and ChIP-Seq data generation process.

In this paper, we demonstrate that the peak pair assumption of Rhee
and Pugh \cite{exo1} does not hold well in real ChIP-exo
data. Furthermore, we found that when we analyze ChIP-exo data from
eukaryotic genomes, it is important to consider sequence biases
inherent to ChIP-exo data, such as mappability and GC content in order
to improve sensitivity and specificity of binding site
identification. dPeak outperforms or performs competitively with
ChIP-exo data analysis such as GEM and MACE. More importantly, when
comparable number of reads is used for both ChIP-exo and ChIP-Seq,
dPeak couple with ChIP-exo data provides resolution comparable to PET
ChIP-Seq and both significantly improve the resolution of protein
binding site identification compared to SET-based analysis with any of
the available methods.

\section{Results and discussion}
\label{sec:results}

\subsection{Data considered}
\label{sec:alldata}



\subsubsection{Deeply sequenced E.Coli $\sigma^{70}$ ChIP-exo and ChIP-Seq data}
\label{sec:ourdata}




\subsection{Characteristics of ChIP exo data}
\label{sec:charac}





\subsection{ChIP exo Quality Control pipeline}
\label{sec:QC}





\section*{Methods}
\label{sec:methods}





\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{chip_exo_paper}

\nocite{exo_gb}


\end{document}
