%\VignetteEngine{knitr::knitr}
\documentclass{bmcart}

\usepackage{xcolor}
\usepackage{url}
\usepackage{amsmath}
\usepackage{amsthm}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{tikz}
\usetikzlibrary{shapes,arrows}
\usepackage{float}
\usepackage{verbatim}

% \usepackage{authblk}
\usepackage[utf8]{inputenc} %unicode support


\begin{document}

%\maketitle

\begin{frontmatter}

\begin{fmbox}
\dochead{Draft}

\title{High Resolution Identification of Protein-DNA Binding
  Events and Quality Control for ChIP-exo data}

\author[
   addressref={aff6},                   % id's of addresses, e.g. {aff1,aff2}
   email={chungd@musc.edu}   % email address
]{\inits{DC}\fnm{Dongjun} \snm{Chung}}
\author[
   addressref={aff1},
   email={welch@stat.wisc.edu}
]{\inits{RW}\fnm{Rene} \snm{Welch}}
\author[
   addressref={aff3},
   email={ong@cs.wisc.edu}
]{\inits{IO}\fnm{Irene} \snm{Ong}}
\author[
   addressref={aff3,aff4},
   email={jagrass@wisc.edu}
]{\inits{JG}\fnm{Jeffrey} \snm{Grass}}
\author[
   addressref={aff3,aff4,aff5},
   email={landick@bact.wisc.edu}
]{\inits{RL}\fnm{Robert} \snm{Landick}}
\author[
   addressref={aff1,aff2},
   corref={aff1},
   email={keles@stat.wisc.edu}
]{\inits{SK}\fnm{S\"und\"uz} \snm{Kele\c{s}}}



\address[id=aff1]{%                           % unique id
  \orgname{Department of Statistics, University of Wisconsin Madison}, % university, etc
  \street{1300 University Avenue},                     %
  %\postcode{}                                % post or zip code
  \city{Madison},                              % city
  \cny{WI}                                    % country
}

\address[id=aff2]{%
  \orgname{Department of Biostatistics and Medical Informatics, University of Wisconsin Madison},
  \street{600 Highland Avenue},
%  \postcode{24105}
  \city{Madison},
  \cny{WI}
}
\address[id=aff3]{
  \orgname{Great Lakes Bioenergy Research Center, University of Wisconsin Madison},
  \street{1552 University Avenue},
  \city{Madison},
  \cny{WI}
}
\address[id=aff4]{
  \orgname{Department of Biochemistry, University of Wisconsin Madison},
  \street{433 Babcock Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff5]{
  \orgname{Department of Bacteriology, University of Wisconsin Madison},
  \street{1550 Linden Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff6]{
  \orgname{Department of Public Health Sciences, Medical University of South Carolina},
  \street{135 Cannon Street},
  \city{Charleston},
  \cny{SC}
}


% \begin{artnotes}
% %\note{Sample of title note}     % note to the article
% \note[id=n1]{Equal contributor} % note, connected to author
% \end{artnotes}


\end{fmbox}

\begin{abstractbox}

  \begin{abstract}

  Recently, ChIP-exo has been developed to investigate protein-DNA
  interaction in higher resolution compared to popularly used
  ChIP-Seq. Although ChIP-exo has drawn much attention and is
  considered as powerful assay, currently, no systematic studies have
  yet been conducted to determine optimal strategies for experimental
  design and analysis of ChIP-exo. In order to address these
  questions, we evaluated diverse aspects of ChIP-exo and found the
  following characteristics of ChIP-exo data. First, Background of
  ChIP-exo data is qute different from that of ChIP-Seq data. However,
  sequence biases inherently present in ChIP-Seq data still exist in
  ChIP-exo data. Second, in ChIP-exo data, reads are located around
  binding sites much more tightly and hence, it has potential for high
  resolution identification of protein-DNA interaction sites, hence
  the space to allocate the reads is greatly reduced. Third, although
  often assumed in the ChIP-exo data analysis methods, the peak pair
  assumption does not hold well in real ChIP-exo data. Fourth, spatial
  resolution of ChIP-exo is comparable to that of PET ChIP-Seq and
  both of them are significantly better than resolution of SET
  ChIP-Seq. Finally, for given fixed sequencing depth, ChIP-exo
  provides higher sensitivity, specificity, and spatial resolution
  than PET ChIP-Seq.

  In this article, we provide a quality control pipeline which
  visually asses ChIP-exo biases and calculates a signal-to-noise
  measure. Also, we updated dPeak \cite{dpeak}, which makes a striking
  balance in sensitivity, specificity, and spatial resolution for
  ChIP-exo data analysis.

\end{abstract}
\begin{keyword}
  \kwd{ChIP-exo}
  \kwd{QC}
  \kwd{TFBS}
  \kwd{BS identification on High-Res}
\end{keyword}

\end{abstractbox}

\end{frontmatter}

\newpage

* to remove later

\tableofcontents


\newpage

\section{Background}
\label{sec:intro}

ChIP-exo (Chromatin Immunopecipitation followed by exonuclease
digestion and next generation sequencing) Rhee and Pugh (\cite{exo1})
is the state-of-the-art experiment developed to attain single
base-pair resolution of protein binding site identification and it is
considered as a powerful alternative to popularly used ChIP-Seq
(Chromatin Immunoprecipitation coupled with next generation sequencing
) assay. ChIP-exo experiments first capture millions of DNA fragments
(150 - 250 bp in length) that the protein under study interacts with
using random fragmentation of DNA and a protein-specific
antibody. Then, exonuclease is introducted to trim 5' end of each DNA
fragment to a fixed distance from the bound protein. As a result,
boundaries around the protein of interest constructed with 5' ends of
fragments are located much closed to bound protein compared to
ChIP-Seq. This is the step unique to ChIP-exo that could potentially
provide significantly higher spatial resolution compared to
ChIP-Seq. Finally, high throughput sequencing of a small region (25 to
100 bp) at 5' end of each fragment generates millions of reads or
tags.

While the number of produced ChIP-exo data keeps increasing,
characteristicos of ChIP-exo data and optimal strategies for
experimental design and analysis of ChIP-exo data are not fully
investigated yet, including issues of sequence biases inherent to
ChIP-exo data, choice of optimal statistical methods, and
determination of optimal sequencing deoth. However, currently, the
number of available ChIP-exo data is still limited and their
sequencing depths are still insufficient for such investigation. To
address this limitation we gathered ChIP-exo data from diverse
organisms: CTCF factor in human \cite{exo1}; ER factor in human and
FoxA1 factor in mouse from \cite{exoillumina}; and generated
$\sigma^{70}$ factor in Escherichia coli (E. Coli) under aerobic ($ +
O_2$) condition, and treated by rifampicin by 0 and 20 minutes.

DNA libraries generated by the ChIP-exo protocol seem to be less
complex than the libraries generated by ChIP-Seq
\cite{exo_review}. Hence, most of current guidelines \cite{encode_qc}
may not be aplicable on ChIP-exo. To address this challenge, we
suggest a series of quality control visualizations to understand which
biases are present in ChIP-exo data. Related to quality control,
Previous ChIP-exo analysis used ChIP-Seq samples to compare the
resolution between experiments (\cite{exo1}, \cite{exo2},
\cite{exoillumina}). In \cite{carroll.qc}, Carroll et. al. studied the
use of the Strand-Cross Correlation (SCC) \cite{strandcc}. and showed
that by filtering blacklisted regions the estimation of the SCC is
improved. However, this method requires to know blacklisted regions in
advance which may not be available. In our pipeline we propose two
out-the-shelf metrics equivalent to RSC and NSC for the estimation of
the signal-to-noise ratio for a ChIP-exo sample.

In order to archieve the potential benefits of ChIP-exo on protein
binding site identification, it is critical to understand which are
the important characteristics of ChIP-exo data and to use algorithms
that could fully utilize information available in ChIP-exo data. Rhee
and Pugh \cite{exo1} discussed that reads in the forward and reverse
strand might construct peak pairs around bound protein, of which
heights were implicitly assumed to be symmetric. Hence, they used the
``peak pair method'' that predicts the midpoint of two modes of peak
pairs as potential binding site. Mace \cite{mace}, CexoR \cite{cexor}
and peakzilla \cite{peakzilla}, recently developed ChIP-exo data
analysis methods, are also based on this peak pair
assumption. However, appropriatness of such assumption was not fully
evaluated in the literature yet.  Furthermore, it is still unknown
which factors could affect protein binding site ientification using
ChIP-exo data. In order to address this problem, we investigated
various aspects of ChIP-exo data by contrasting them with their
respective ChIP-Seq experiments.

Currently, research on statistical methods for ChIP-exo data is still
in its very early stage. Although many methods have been proposed to
identify protein binding sites from ChIP-Seq data (reviewed in
\cite{evaluation} and \cite{computation}), such as MACS \cite{macs},
CisGenome \cite{cisgenome} and MOSAiCS \cite{mosaics}, these
approaches reveal protein binding sites in lower resolution, i.e., at
an interval of hundreds to thousands of base pairs. Furthermore, they
report only one ``mode'' or ``predicted binding location'' per
peak. Hence, these methods are not appropiate to evaluate the
potential of ChIP-exo data for high resolution identification of
protein binding sites. More recently, deconvolution algorithms such as
CSDeconv \cite{csdeconv}, GEM \cite{gem} (an improved version of
\cite{gps}) and PICS \cite{pics} have been proposed to identify
binding sites in higher resolution using ChIP-Seq data. However, most
of them are still not tailored for ChIP-exo and PET and SET ChIP-Seq
data in a unified framework and as a result, currently available
methods are not appropiate for fair comparison between ChIP-exo and
ChIP-Seq. To address these limitations, we developed an improved of
dPeak \cite{dpeak}, a high resolution binding site identification
(deconvolution) algorithm that we previously developed for PET and SET
ChIP-Seq data, so that it can also handle ChIP-exo data. The dPeak
algorithm implements a probabilistic model that accurately describes
the ChIP-exo and ChIP-Seq data generation process.

In this paper, we demonstrate that the peak pair assumption of Rhee
and Pugh \cite{exo1} does not hold well in real ChIP-exo
data. Furthermore, we found that when we analyze ChIP-exo data from
eukaryotic genomes, it is important to consider sequence biases
inherent to ChIP-exo data, such as mappability and GC content in order
to improve sensitivity and specificity of binding site
identification. dPeak outperforms or performs competitively with
ChIP-exo data analysis such as GEM and MACE. More importantly, when
comparable number of reads is used for both ChIP-exo and ChIP-Seq,
dPeak couple with ChIP-exo data provides resolution comparable to PET
ChIP-Seq and both significantly improve the resolution of protein
binding site identification compared to SET-based analysis with any of
the available methods.

\section{Results and discussion}
\label{sec:results}

\subsection{Deeply sequenced E.Coli $\sigma^{70}$ ChIP-exo and ChIP-Seq data}
\label{sec:ourdata}

$\sigma^{70}$ factor is a transcription initiation factor of
housekeeping genesin E. coli. In E. coli genomes, many promoters
contain multiple transcription start sites (TSS) and these TSS are
often closely space (10 ~ 150 bp). These closely spaced binding sites
are considered to be multiple ``switches'' that differentially
regulate gene expression under diverse growth conditions
\cite{regdb_old}, \cite{regulondb}. Therefore, investigation of
ChIP-exo's potential for identification and differentiation of closely
spaced binding sites are invaluable for elucidating the
transcriptional networks of prokaryotic genomes.

\subsection{Current guidelines and quality metrics}
\label{sec:qcnow}

To have a first view on the data, we calculated a series of the most
commonly used quality indicators for ChIP-Seq data, which are listed
in table \ref{tab:qcbase}. The PBC, NSC and RSC were calculated as
defined in \cite{encode_qc} and the SSD was calculated as defined in
\cite{htseq}. Additionally, we calculated the FSR to compare the
number of fragments in each strand. A few conclussions that we may get
by observing the PBC in table \ref{tab:qcbase} are all the samples
have a quite low library complexity, hence it would be suggested to
repeat the experiment. 

\begin{table}[h]
  \centering
\begin{tabular}{l|r|r|r|r|r|r|r}
\hline\hline
Condition & Rep. & Nr. reads & PBC & SSD & FSR & NSC & RSC \\
\hline\hline
Aerobic & 1 & 1,454,566 & 0.2156 & 0.0830 & 0.5012 & 114.14 & -1.670\\
\hline
Aerobic & 2 & 864,714 & 0.2471 & 0.0488 & 0.4923 & 160.93 & -1.503 \\
\hline
Rif-0min & 1 & 960,256 & 0.2823 & 0.0361 & 0.4998 & 10.29 & -0.092\\
\hline
Rif-0min & 2 & 2,247,295 & 0.2656 & 0.1091 & 0.5047 &  25.08 & -0.2279 \\
\hline
Rif-20min & 1 & 1,940,387 & 0.2698 & 0.0820 & 0.5054 & 17.69 & -0.450 \\
\hline
Rif-20min & 2 & 4,229,574 & 0.2153 & 0.1647 & 0.5071 &  14.11 & -0.533\\
\hline
\end{tabular}  
\caption{Usual quality control indicators applied to our $\sigma^{70}$ samples. PBC stands PCR-bottleneck coefficient, SSD for standarized standard deviation, FSR for forward strand ratio, NSC for normalized strand cross-correlation coefficient and RSC for relative strand cross-correlation coefficient.}
  \label{tab:qcbase}
\end{table}



\subsection{ChIP-exo Quality Control pipeline}
\label{sec:QC}

Figure \ref{fig:qcdiagram} shows a flowchart for the ChIPexoQC
pipeline. Which basically partitions genome by keeping the
non-diggested ChIP-exo regions. Then, for each region it calculates a
series of independent statistics. Finally, it creates several
visualizations that were designed to asses the overall quality levels
of A ChIP-exo sample.

\begin{figure}[h!]
  \centering
  \includegraphics[width = .9\textwidth]{../figs/for_paper/coverage_diagram.pdf}
  \caption{Diagram of the ChIP-exo Quality control pipeline(this is a
    place-holder). The genome is partitioned into several regions by
    removing the empty segments and for each regions we calculate a
    collection of qc indicators.}
  \label{fig:qcdiagram}
\end{figure}



\subsubsection{Assessing the strand imbalance in ChIP-exo data}
\label{sec:strand_imbalance}

The strand imbalance assesment is based in the observation that the
enriched regions usually have a higher concentration of fragments,
therefore we examine the FSR (defined as the ratio of the number of
forward stranded reads divided by the total number of reads in a given
region) as the region with lower depth are being filtered out. This
indicator is of particular importance, since several methods rely on
the ``peak-pair'' assumption. In table \ref{tab:qcbase}, we calculated
the FSR and noticed that for all the samples, it's value is close to
0.5, which means that there are roughly the same amount of reads in
both strands. However, as we can see in figure \ref{fig:fsr}



\begin{figure}[h]
  \centering
 
  \caption{This is a place-holder figure by now. The idea of this segments it }
  \label{fig:fsr}
\end{figure}



\subsection{Comparison with ChIP-Seq data}
\label{sec:comp}

We compared 

\begin{figure}[h]
  \centering
  \includegraphics[width = .6\textwidth,page = 3 ]{../figs/for_paper/ChIPseqPET_ChIPexo_tagCount_comparison.pdf}
  \includegraphics[width = .9\textwidth ]{../figs/for_paper/MA_plot_fwd_bwd_by_seq.pdf}
  \caption{Comparison}
  \label{fig:comp}
\end{figure}



% \[
% \mbox{FSR} = \frac{\mbox{Number of forward strand reads in region}}{\mbox{Total number of reads in region}}
% \]







\subsection{Recomendations for the design of ChIP-exo experiments}
\label{sec:reco}

<<design_param , include = FALSE , echo = FALSE , eval = TRUE>>=

  K <- 10
  ext <- 15
  
@ 

We sampled a fixed amoung of fragments for each of the ChIP-exo, PET
ChIP-Seq and SET ChIP-Seq datasets of the $\sigma^{70}$ sample in
aerobic conditions. For each sampled dataset we applied our
lower-to-higher resolution pipeline by calling peaks with MOSAiCS
\cite{mosaics} and then deconvoling the binding events by using dPeak
\cite{dpeak}. For the ChIP-exo datasets we called peaks by using
GC-content and mappability with MOSAiCS, and for the ChIP-Seq datasets
we used their respective Input samples.

\begin{figure}[h!]
  \centering
  \includegraphics[width = .8\textwidth]{../figs/for_paper/Sig70_aerobic_saturation.pdf}
  \caption{Comparison of the number of candidate regions (A),
    predicted events (B), identified targets (C) and resolution (D)
    among ChIP-exo, PET ChIP-Seq and SET ChIP-Seq. RegulonDB
    annotations are considered as a gold standard. A gold standard was
    considered identified if a BS was estimated inside a \Sexpr{ext}
    bp window around it}
  \label{fig:design}
\end{figure}

In figure \ref{fig:design} we compared the behaviour of each data type
when their depth is fixed. It is remarkable that even when the number
of candiate peaks or the number of predicted events in quite lower
than for both ChIP-Seq cases, it outperforms them in the number of
identified targets and in resolution. This may suggest that with
ChIP-exo less false positive peaks are being called and that when the
targets are being identified, dpeak is able to estimate binding
locations closer to the actual true cases. Additionally, we can see
that as the read depth increases all four indicators do so as well,
which may indicate that with ChIP-exo a smaller amount of reads is
necessary to identify a higher number of targets, but it may also be
possible that this is an artifact occurring due to ChIP-exo's lower
library complexity.



\newpage


\section{Methods}
\label{sec:methods}

\subsection{Growth conditions}
\label{sec:growth}


\subsection{ChIP-exo experiments}
\label{sec:experiments}



\subsection{Library preparation, sequencing and mapping of sequencing reads}
\label{sec:library}

\subsection{Method comparison for ChIP-exo}
\label{sec:method_comp}






\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{chip_exo_paper}

\nocite{exo_gb}


\end{document}
