%\VignetteEngine{knitr::knitr}
\documentclass{bmcart}

\usepackage{xcolor}
\usepackage{url}
\usepackage{amsmath}
\usepackage{amsthm}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{tikz}
\usetikzlibrary{shapes,arrows}
\usepackage{float}
\usepackage{verbatim}

% \usepackage{authblk}
\usepackage[utf8]{inputenc} %unicode support


\begin{document}

%\maketitle

\begin{frontmatter}

\begin{fmbox}
\dochead{Draft}

\title{High Resolution Identification of Protein-DNA Binding Events
  using ChIP-exo}

\author[
   addressref={aff6},                   % id's of addresses, e.g. {aff1,aff2}
   email={chungd@musc.edu}   % email address
]{\inits{DC}\fnm{Dongjun} \snm{Chung}}
\author[
   addressref={aff1},
   email={welch@stat.wisc.edu}
]{\inits{RW}\fnm{Rene} \snm{Welch}}
\author[
   addressref={aff3},
   email={ong@cs.wisc.edu}
]{\inits{IO}\fnm{Irene} \snm{Ong}}
\author[
   addressref={aff3,aff4},
   email={jagrass@wisc.edu}
]{\inits{JG}\fnm{Jeffrey} \snm{Grass}}
\author[
   addressref={aff3,aff4,aff5},
   email={landick@bact.wisc.edu}
]{\inits{RL}\fnm{Robert} \snm{Landick}}
\author[
   addressref={aff1,aff2},
   corref={aff1},
   email={keles@stat.wisc.edu}
]{\inits{SK}\fnm{S\"und\"uz} \snm{Kele\c{s}}}



\address[id=aff6]{
  \orgname{Department of Public Health Sciences, Medical University of South Carolina},
  \street{135 Cannon Street},
  \city{Charleston},
  \cny{SC}
}
\address[id=aff1]{%                           % unique id
  \orgname{Department of Statistics, University of Wisconsin Madison}, % university, etc
  \street{1300 University Avenue},                     %
  %\postcode{}                                % post or zip code
  \city{Madison},                              % city
  \cny{WI}                                    % country
}

\address[id=aff2]{%
  \orgname{Department of Biostatistics and Medical Informatics, University of Wisconsin Madison},
  \street{600 Highland Avenue},
%  \postcode{24105}
  \city{Madison},
  \cny{WI}
}
\address[id=aff3]{
  \orgname{Great Lakes Bioenergy Research Center, University of Wisconsin Madison},
  \street{1552 University Avenue},
  \city{Madison},
  \cny{WI}
}
\address[id=aff4]{
  \orgname{Department of Biochemistry, University of Wisconsin Madison},
  \street{433 Babcock Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff5]{
  \orgname{Department of Bacteriology, University of Wisconsin Madison},
  \street{1550 Linden Drive},
  \city{Madison},
  \cny{WI}
}


% \begin{artnotes}
% %\note{Sample of title note}     % note to the article
% \note[id=n1]{Equal contributor} % note, connected to author
% \end{artnotes}




\end{fmbox}

\begin{abstractbox}

  \begin{abstract}

  Recently, ChIP-exo has been developed to investigate protein-DNA
  interaction in higher resolution compared to popularly used
  ChIP-Seq. Although ChIP-exo has drawn much attention and is
  considered as powerful assay, currently, no systematic studies have
  yet been conducted to determine optimal strategies for experimental
  design and analysis of ChIP-exo. In order to address these
  questions, we evaluated diverse aspects of ChIP-exo and found the
  following characteristics of ChIP-exo data. First, Background of
  ChIP-exo data is qute different from that of ChIP-Seq data. However,
  sequence biases inherently present in ChIP-Seq data still exist in
  ChIP-exo data. Second, in ChIP-exo data, reads are located around
  binding sites much more tightly and hence, it has potential for high
  resolution identification of protein-DNA interaction sites, hence
  the space to allocate the reads is greatly reduced. Third, although
  often assumed in the ChIP-exo data analysis methods, the peak pair
  assumption does not hold well in real ChIP-exo data. Fourth, spatial
  resolution of ChIP-exo is comparable to that of PET ChIP-Seq and
  both of them are significantly better than resolution of SET
  ChIP-Seq. Finally, for given fixed sequencing depth, ChIP-exo
  provides higher sensitivity, specificity, and spatial resolution
  than PET ChIP-Seq.

  In this article, we provide a quality control pipeline which
  visually asses ChIP-exo biases and calculates a signal-to-noise
  measure. Also, we updated dPeak, which makes a striking
  balance in sensitivity, specificity, and spatial resolution for
  ChIP-exo data analysis.

\end{abstract}
\begin{keyword}
  \kwd{ChIP-exo}
  \kwd{Quality control}
\end{keyword}

 
\end{abstractbox}

\end{frontmatter}

\newpage

* to remove later

\tableofcontents




\newpage


\section{Background}
\label{sec:intro}

ChIP-exo (Chromatin Immunopecipitation followed by exonuclease
digestion and next generation sequencing) Rhee and Pugh (\cite{exo1})
is the state-of-the-art experiment developed to attain single
base-pair resolution of protein binding site identification and it is
considered as a powerful alternative to popularly used ChIP-Seq
(Chromatin Immunoprecipitation coupled with next generation sequencing
) assay. ChIP-exo experiments first capture millions of DNA fragments
(150 - 250 bp in length) that the protein under study interacts with
using random fragmentation of DNA and a protein-specific
antibody. Then, exonuclease is introducted to trim 5' end of each DNA
fragment to a fixed distance from the bound protein. As a result,
boundaries around the protein of interest constructed with 5' ends of
fragments are located much closed to bound protein compared to
ChIP-Seq. This is the step unique to ChIP-exo that could potentially
provide significantly higher spatial resolution compared to
ChIP-Seq. Finally, high throughput sequencing of a small region (25 to
100 bp) at 5' end of each fragment generates millions of reads or
tags.

While the number of produced ChIP-exo data keeps increasing,
characteristicos of ChIP-exo data and optimal strategies for
experimental design and analysis of ChIP-exo data are not fully
investigated yet, including issues of sequence biases inherent to
ChIP-exo data, choice of optimal statistical methods, and
determination of optimal sequencing deoth. However, currently, the
number of available ChIP-exo data is still limited and their
sequencing depths are still insufficient for such investigation.

\section{Results and discussion}
\label{sec:results}







\section*{Methods}
\label{sec:methods}






\bibliographystyle{bmc-mathphys} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{chip_exo_paper}

\nocite{exo_gb}

\end{document}
