\documentclass[11pt]{article}
%\documentclass[11pt]{scrartcl}
\usepackage{amsmath, amsfonts, amssymb}
\usepackage{adjustbox}
\usepackage{fullpage}
\usepackage{epsfig}
%\setkomafont{disposition}{\normalfont\bfseries}
% \usepackage[round, sort]{natbib}
\renewcommand{\baselinestretch}{1.4}
\usepackage{float}


\title{High Resolution Identification of Protein-DNA Binding Events
  and Quality Control for ChIP-exo data\vspace*{\fill}}

\author{Rene Welch\\Preliminary Examination\\Department of Statistics, University of Wisconsin-Madison}

\date{December 1st, 2015}

%\usepackage{Sweave}
\begin{document}

\maketitle

\vspace*{\fill}

\textbf{Committee Members:}

\textbf{Professor S\"und\"uz Kele\c{s}}, Department of Statistics,
Department of Biostatistics and Medical Informatics

\textbf{Professor Karl Broman}, Department of Biostatistics and
Medical Informatics

\textbf{Professor Colin Dewey}, Department of Computer Sciences,
Department of Biostatistics and Medical Informatics

\textbf{Professor Christina Kendziorski}, Department of Biostatistics
and Medical Informatics

\textbf{Professor Ming Yuan}, Department of Statistics

\thispagestyle{empty}


\newpage

\tableofcontents

\newpage

\section*{Abstract}

    Recently, ChIP-exo has been developed to investigate protein-DNA
    interaction in higher resolution compared to popularly used
    ChIP-Seq. Although ChIP-exo has drawn much attention and is
    considered as powerful assay, currently, no systematic studies
    have yet been conducted to determine optimal strategies for
    experimental design and analysis of ChIP-exo. In order to address
    these questions, we evaluated diverse aspects of ChIP-exo and
    found the following characteristics of ChIP-exo data. First, the
    background of ChIP-exo data is quite different from that of
    ChIP-Seq data. However, sequence biases inherently present in
    ChIP-Seq data still exist in ChIP-exo data. Second, in ChIP-exo
    data, reads are located around binding sites much more tightly and
    hence, it has potential for high resolution identification of
    protein-DNA interaction sites, and also the space to allocate the
    reads is greatly reduced. Third, although often assumed in the
    ChIP-exo data analysis methods, the ``peak pair'' assumption does
    not hold well in real ChIP-exo data. Fourth, spatial resolution of
    ChIP-exo is comparable to that of PET ChIP-Seq and both of them
    are significantly better than resolution of SET ChIP-Seq. Finally,
    for given fixed sequencing depth, ChIP-exo provides higher
    sensitivity, specificity and spatial resolution than PET
    ChIP-Seq.

    In this article we provide a quality control pipeline which
    visually asses ChIP-exo biases and calculates a signal-to-noise
    measure. Also, we updated dPeak \cite{dpeak}, which makes a
    striking balance in sensitivity, specificity and spatial
    resolution for ChIP-exo data analysis.


\section{Introduction}
\label{sec:intro}

\section{Statistical Framework for ChIP-exo}
\label{sec:stat}



\section{Results}
\label{sec:results}



\section{Planned work}
\label{sec:future}





\newpage

\bibliographystyle{plain} % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliography{chip_exo_paper}

\end{document}
