
## PCR bottleneck report

### PCR bottleneck		

```{r package_options, include=FALSE}
opts_knit$set(progress = TRUE, verbose = TRUE)
```

According to the **ENCODE** definitions of [quality metrics](http://encodeproject.org/ENCODE/qualityMetrics.html#definitions), one of the metrics used is the *PCR bottleneck coefficient* or **PBC**.

The definition is:

A measure of library complexity, i.e. how skewed the distribution of read counts per location is towards 1 read per location.

PBC = N1/Nd

(where N1= number of genomic locations to which **EXACTLY** one unique mapping read maps, and Nd = the number of genomic locations to which **AT LEAST** one unique mapping read maps, i.e. the number of non-redundant, unique mapping reads).

In particular we can see that always N1 <= Nd, so 0<= PBC <= 1. ENCODE recomends the following classification:

| PBC range | Bottleneck class |
| :---:     | :---: |
|0 -0.5  | Severe |
| 0.5-0.8 | Moderate|
|0.8-0.9 | Mild |
|0.9 - 1| Non-existant|

```{r init,include = FALSE,echo = TRUE,results = "hide"}
library(parallel)
library(GenomicAlignments)
library(ggplot2)
dr ="/NO_BACKUP/KelesGroup_DongjunChung/ChIP-exo/Landick_ChIP-exo3/rawdata"
folder = c("ChIPexo","ChIPseq_PET","ChIPseq_SET")
files = lapply(folder,function(x,dr){
  ff = list.files(file.path(dr,x))
  return(ff[!grepl("bai",ff) & !grepl("sam",ff)])},dr)
names(files) = folder
```
### Code

The following function was used to calculate the PBC:

```{r PBC_fun,include = TRUE,results = "hide",echo = TRUE}
PBC <- function(file)
{
  require(GenomicAlignments)
  rr = readGAlignmentsFromBam(file,param = NULL)
  ss1 = subset(rr,subset =strand(rr)=="+")
  ss2 = subset(rr,subset = strand(rr)=="-")
  N11 = length(unique(start(ss1)))
  N12 = length(unique(end(ss2)))
  N1 = N11+N12
  Nd1 = length(ss1)
  Nd2 = length(ss2)
  Nd = Nd1 + Nd2
  PBC1 = round(N11 / Nd1,4)
  PBC2 = round(N12 / Nd2,4)
  PBC = round(N1 /Nd,4)
  return(c(PBC_plus=PBC1,PBC_minus=PBC2,PBC=PBC))  
}
```

```{r load_data,include = FALSE,echo = TRUE,results = "hide"}
load("../RData/PBC.RData")
tab = list()
tab[[1]] = as.data.frame(t(do.call(cbind,SET)))
tab[[1]]$type = "ChIP-Seq.SET"
tab[[2]] = as.data.frame(t(do.call(cbind,PET)))
tab[[2]]$type = "ChIP-Seq.PET"
tab[[3]] = as.data.frame(t(do.call(cbind,EXO)))
tab[[3]]$type = "ChIP-Exo"
```
### Figures

#### Figure 1

```{r overall_boxplot,include = TRUE,echo = FALSE,results = "hide",fig.height = 4,fig.width = 4}
df = do.call(rbind,tab)
df$type = factor(df$type)
p <- ggplot(df,aes(type,PBC))+geom_boxplot()+
   theme(legend.position = "none",axis.text.x = element_text(angle = 90))
print(p)
```

We can see:
- The overall level of the ChIP-Exo's PBC coefficients is lower than the level of the other two ChIP-Seq labels
- The range of the ChIP-Exo's PBC coefficients is also smaller than other two ranges. In particular, the interquartile range of the PET coefficients is smaller than the interquartile range of the SET coefficients.

#### Figure 2

```{r boxplot_strand,include = TRUE,echo = FALSE,results = "hide",fig.height = 4,fig.width = 12}
tab1 = lapply(tab[3:1],function(x)rbind(data.frame(type = x$type,PBC = x$PBC_plus,strand = "+"),
  data.frame(type = x$type,PBC = x$PBC_minus,strand = "-")))
df1 = do.call(rbind,tab1)
df1$type = factor(df1$type)
df1$strand = factor(df1$strand)
p1 <- ggplot(df1,aes(strand,PBC))+geom_boxplot()+facet_grid(.~type)+
  theme(legend.position = "none")
print(p1)
```

- For this plot, the PBC coefficents was calculated using only the reads with certain strand. In particular, we can see that for all 3 types, the boxplots for both strands are very similar.
- The observations of figure 1 hold for both strands.



### Tables

#### ChIP - Seq - SET
```{r show_set,include = TRUE,echo = FALSE, results = "asis"}
kable(tab[[1]][,1:3],format = "markdown")
```

#### ChIP - Seq - PET
```{r show_pet,include = TRUE,echo = FALSE, results = "asis"}
kable(tab[[2]][,1:3],format = "markdown")
```

#### ChIP - Exo
```{r show_exo,include = TRUE,echo = FALSE, results = "asis"}
kable(tab[[3]][,1:3],format = "markdown")
```

