\documentclass{bmcart}

\usepackage{xcolor}
\usepackage{url}
\usepackage{amsmath}
\usepackage{amsthm}
\usepackage{amssymb}
\usepackage{graphicx}
\usepackage{tikz}
\usetikzlibrary{shapes,arrows}
\usepackage{float}
\usepackage{verbatim}
\usepackage{tabu}
\usepackage{multirow}
\usepackage{framed}

\usepackage{soul}

\newcommand{\beginsupplement}{%
        \setcounter{table}{0}
        \renewcommand{\thetable}{S\arabic{table}}%
        \setcounter{figure}{0}
        \renewcommand{\thefigure}{S\arabic{figure}}%
     }

\newcommand{\pname}[1]{\texttt{ChIPexoQual}}
\newcommand{\SK}[1]{\textcolor{red}{SK: #1}}

% \usepackage{authblk}
\usepackage[utf8]{inputenc} %unicode support

\newcommand{\sig}{\sigma^{70}}

\begin{document}

\begin{frontmatter}

\begin{fmbox}
\dochead{Draft}

\title{Data Exploration, Quality Control, and Statistical Analysis of
  ChIP-exo/nexus Experiments}

\author[
   addressref={aff1},
   noteref={n1},
   email={welch@stat.wisc.edu}      
]{\inits{RW}\fnm{Rene} \snm{Welch} }
\author[
   addressref={aff6},        % id's of addresses, e.g. {aff1,aff2}
   noteref={n1},
   email={chungd@musc.edu}   % email address
]{\inits{DC}\fnm{Dongjun} \snm{Chung} } 
\author[
   addressref={aff3},
   email={ong@cs.wisc.edu}
]{\inits{IO}\fnm{Irene} \snm{Ong}}
\author[
   addressref={aff3,aff4},
   email={jagrass@wisc.edu}
]{\inits{JG}\fnm{Jeffrey} \snm{Grass}}
\author[
   addressref={aff3,aff4,aff5},
   email={landick@bact.wisc.edu}
]{\inits{RL}\fnm{Robert} \snm{Landick}}
\author[
   addressref={aff1,aff2},
   corref={aff1},
   email={keles@stat.wisc.edu}
]{\inits{SK}\fnm{S\"und\"uz} \snm{Kele\c{s}}}

\address[id=aff1]{%                           % unique id
  \orgname{Department of Statistics, University of Wisconsin Madison}, % university, etc
  \street{1300 University Avenue},                     %
  %\postcode{}                                % post or zip code
  \city{Madison},                              % city
  \cny{WI}                                    % country
}

\address[id=aff2]{%
  \orgname{Department of Biostatistics and Medical Informatics, University of Wisconsin Madison},
  \street{600 Highland Avenue},
%  \postcode{24105}
  \city{Madison},
  \cny{WI}
}
\address[id=aff3]{
  \orgname{Great Lakes Bioenergy Research Center, University of Wisconsin Madison},
  \street{1552 University Avenue},
  \city{Madison},
  \cny{WI}
}
\address[id=aff4]{
  \orgname{Department of Biochemistry, University of Wisconsin Madison},
  \street{433 Babcock Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff5]{
  \orgname{Department of Bacteriology, University of Wisconsin Madison},
  \street{1550 Linden Drive},
  \city{Madison},
  \cny{WI}
}
\address[id=aff6]{
  \orgname{Department of Public Health Sciences, Medical University of South Carolina},
  \street{135 Cannon Street},
  \city{Charleston},
  \cny{SC}
}

\begin{artnotes}
\note[id=n1]{These two authors contributed equally.} % note, connected to author
\end{artnotes}


\end{fmbox}

\begin{abstractbox}

  \begin{abstract}
ChIP-exo/nexus experiments present  modifications on the commonly used ChIP-seq protocol
    for high
    resolution mapping of transcription factor binding sites. Although
    many aspects of the ChIP-exo data analysis are similar to those of
    ChIP-seq, these high throughput experiments present a number of unique quality control and analysis challenges. We
    develop a statistical quality control pipeline and accompanying \texttt{R} package, \pname{}, 
    to enable exploration and analysis of ChIP-exo and related experiments. \pname{}  evaluates a number of key
    issues including strand imbalance, library complexity, and
    signal enrichment of data from ChIP-exo/nexus experiments. Assessment of these properties  are facilitated
    through diagnostic plots and summary statistics calculated over
    regions of the genome with varying levels of coverage.
    %\pname pipeline explores and quantifies these aspects by partitioning the experiment reads into a collection of regions, calculating a series of summary statistics for each region, providing visualizations and calculating measures to globally asses t quality of a ChIP-exo experiment. 
    
    We evaluated our QC pipeline with both  large collections of public ChIP-exo/nexus data and 
    multiple, new ChIP-exo datasets from \textit{E. Coli}.  \pname{} analysis of these datasets resulted in  guidelines  for
   using these QC metrics across a wide range of sequencing depths 
   and further insights for modeling ChIP-exo data. %In addition, we demonstrate that the
    %ChIP-exo QC pipeline it is also applicable to ChIP-nexus data,
    %showing that those experiments present higher quality than
    %ChIP-exo experiments under similar conditions.
    Finally, although ChIP-exo experiments have been compared to ChIP-seq experiments with single-end (SE) sequencing, we provide, for the first time, comparisons with paired-end (PE)  ChIP-seq experiments.
    %We compared ChIP-exo with Paired End (PE) and Single End (SE)
    %ChIP-Seq and found the following characteristics: First, although
    %often assumed in ChIP-exo data analysis methods, the ``peak pair''
    %assumptions does not hold locally in actual ChIP-exo data. Second,
    %we for the first time compared PE ChIP-Seq with ChIP-exo and found
%    that 
    We illustrate that, at fixed sequencing depths, ChIP-exo provides higher
    sensitivity, specificity, and spatial resolution than PE
    ChIP-seq and both significantly outperform their SE ChIP-seq counterpart. Furthermore, we show that ChIP-exo and PE ChIP-seq are
    comparable in sensitivity for closely located binding events, but
    as the average distance between binding events increases, ChIP-exo
    exhibits higher sensitivity than PE ChIP-Seq. \SK{The last sentence does not really make much sense? }

  \end{abstract}

\begin{keyword}
  \kwd{ChIP-exo}
  \kwd{ChIP-nexus}
  \kwd{ChIP-seq}
  \kwd{Statistical Quality Control}
  \kwd{Spatial Resolution}
  \kwd{Transcription Factor}  
  \kwd{Binding Site Identification with High-Resolution}
  \kwd{Deconvolution}
\end{keyword}

\end{abstractbox}

\end{frontmatter}

\newpage

\section*{Background}
\label{sec:intro}

Chromatin Immunoprecipitation followed by exonuclease
digestion and next generation sequencing (ChIP-exo) 
is currently one of the state-of-the-art high throughput assays for profiling protein-DNA interactions at or close to
single base-pair resolution \cite{exo1}. It presents a powerful alternative to popularly used ChIP-seq
(Chromatin immunoprecipitation coupled with next generation
sequencing) assay. ChIP-exo experiments first capture millions of DNA
fragments ($150$ - $250$ bps in length) that the protein under study
interacts with using  a
protein-specific antibody and random fragmentation of DNA. Then, $\lambda$-exonuclease ($\lambda$-exo) is deployed to trim the
$5^{\prime}$ end of each DNA fragment to each protein-DNA interaction boundary. This step is unique to ChIP-exo and
aims to achieve  significantly higher spatial resolution
compared to ChIP-seq. Finally, high throughput sequencing of a small
region ($36$ to $100$ bps) at the $5^{\prime}$ end of each fragment generates
millions of reads. 
 Similarly, ChIP-nexus (Chromatin Immunoprecipitation followed by exonuclease
digestion, unique barcode, single ligation and next generation
ligation)  \cite{chipnexus} is a further modification on the
ChIP-exo protocol. ChIP-nexus aims to overcome limitations of ChIP-exo by 
by yielding high complexity libraries with numbers of cells comparable to that of ChIP-seq experiments. This is achieved by reducing the numbers of ligations in the standard ChIP-exo protocol from two to one, and adding unique, randomized barcodes to adoptors to enable monitoring of overamplification. 
%where both sequencing adaptors are ligated at the
%end of the ChIP fragments. Then, after exonuclease digestion, DNA self-circularization with circLigase, and restriction enzyme cutting between the two adaptors, the final library is amplified. Compared to ChIP-exo, ChIP-nexus aims to   attain  higher resolution analysis while yielding higher complexity libraries.
Figure~\ref{fig:chip_diagram} illustrates the
differences between different ChIP-based protocols: ChIP-exo, single-end (SE) ChIP-seq, paired-end
(PE) ChIP-seq, ChIP-nexus. The $5^{\prime}$ ends of a ChIP-exo/nexus experiment are clustered
more tightly around the binding sites of the protein than in a ChIP-seq
experiment. In a PE ChIP-seq experiment, both ends are sequenced as opposed to only the $5^{\prime}$ end
in a SE ChIP-seq. \SK{Note to Rene: ChIP-Nexus paper has a good description of what can go wrong with ChIP-exo; It seems like most of the read imbalance could also be due to ligation inefficiency. Although this probably does not explain why we have reads only from one strand in some regions. This is more likely explained by over digestion in one strand or single stranded TF-DNA interaction. Think a bit more on these issues and understand how the ligation could give rise to strand imbalance, we can ask Bob to have a closer look at the section where we have this discussion.}

Although ChIP-exo/nexus protocols are being adopted by  research community, features of 
ChIP-exo data, especially those pertaining to data quality,  have not been investigated much. The key features of ChIP-exo/nexus that separate them from ChIP-seq are broadly as follows. First, 
DNA libraries
generated by the ChIP-exo protocol are expected to be less complex than the
libraries generated by ChIP-seq \cite{exo_review})  because digestion by $\lambda$-exo   is expected to restrict the space of genomic positions that sequencing reads can map to, to small local regions around the actual binding sites. Therefore, in  high quality and especially deeply sequenced ChIP-exo datasets, it is possible to observe large numbers of reads accumulating at a small number of bases due to actual signal rather than overamplification bias as commonly observed in ChIP-seq experiments.
Second, although we expect approximately the same
numbers  of reads from  both DNA strands at a given binding site,  there may be locally more reads in
one strand than in the other, owing to $\lambda$-exo efficiency, ligation efficiency, or other factors. 
This is a key point with implications on the statistical analysis of ChIP-exo data. Specifically, 
currently available ChIP-exo specific statistical analysis methods   (e.g., 
 Mace \cite{mace}, CexoR \cite{cexor}, and   Peakzilla \cite{peakzilla}) rely on the existence of 
peak-pairs formed by forward and reverse strand reads at the binding site. 
Finally, most of current widely used ChIP-seq
quality control (QC) guidelines  \cite{encode_qc}
may not be directly applicable to ChIP-exo data.

To address these challenges, we develop 
a suite of diagnostic plots and summary statistics and implement them in a versatile \texttt{R package} named \pname{}.
We apply this pipeline on  a large collection of public and newly generated ChIP-exo/nexus data. We validate implications of the QC pipeline  by  evaluation of the samples for features that capture high signal to noise such as occurrences of   motifs recognized by the profiled DNA interacting protein. Our analyses of this large collection of data revealed that  the so-called ChIP-exo peak-pair assumption is subject to violations. To further address this and provide a platform where ChIP-exo and  ChIP-seq experiments can be evaluated with comparable methods, we assess performances of recently developed methods suitable for ChIP-exo analysis, including dpeak \cite{dpeak} and GEM \cite{gem}. We observe that dPeak performs as good or better than the available ChIP-exo methods and provides a platform where PE and SE ChIP-seq can be compared with their ChIP-exo counterpart. Our comparisons of PE ChIP-seq with ChIP-exo interestingly highlights that while ChIP-exo outperforms PE ChIP-seq in terms of resolution and detection power, both are significantly better than SE ChIP-seq.


% while there are not established QC pipelines for ChIP-exo; previous ChIP-exo analyses used ChIP-Seq samples to compare the resolution between experiments \cite{exo1,exoillumina,exo2}. 

%to interrogate these
%biases in a ChIP-exo experiment and globally asses the enrichment and
%library complexity of a ChIP-exo sample and a procedure to distinguish
%low complexity libraries from deeply sequenced experiments. 
%This
%aspect is unique to ChIP-exo, since the exonuclease enzyme it is
%expected to digested the reads that are not bound to a transcription
%factor, therefore the number of bases where a ChIP-exo fragment could
%be potentially aligned is reduced. That way, in a high quality
%ChIP-exo experiment is possible to observe a large amount of reads to
%be aligned to unique position due to genomic signal instead of PCR
%artifacts.



%In order to obtain the potential benefits of ChIP-exo on protein
%binding site identification, it is critical to use algorithms that
%could fully utilize information available in ChIP-exo data. Rhee and
%Pugh, 2011 \nocite{exo1} discussed that reads in the forward and reverse
%strand might construct peak pairs around bound proteins, of which
%heights were implicitly assumed to be symmetric. Based on this
%rationale, they used the ``peak pair method'' that predicts the
%midpoint of two modes of peak pairs as potential binding
%sites. Recently developed ChIP-exo data analysis methods, such as Mace
%(Wang et al, 20114\nocite{mace}), CexoR (Madrigal, 2015\nocite{cexor})
%and Peakzilla (Bardet et al., 2013\nocite{peakzilla}), are also based
%on this peak pair assumption. However, appropriateness of such
%assumption was not fully evaluated in the literature yet. Furthermore,
%it is still unknown which factors could affect protein binding site
%identification using ChIP-exo data. In order to address this problem,
%we investigated various aspects of ChIP-exo data by contrasting them
%with their respective ChIP-Seq experiments.



%Currently, research on statistical methods for ChIP-exo data is still
%in its very early stage. Although many methods have been proposed to
%identify protein binding sites from ChIP-Seq data (reviewed by
%Wilbanks and Facciotti, 2012\nocite{evaluation} and Pepke and Wold,
%2009 \nocite{computation}), such as MACS (Zhang et al.,
%2008\nocite{macs}), CisGenome (Ji et al., 2008\nocite{cisgenome}) and
%MOSAiCS (Kuan et al., 2009\nocite{mosaics}), these approaches might
%not fully utilize potentials of ChIP-exo data for high resolution
%identification of protein binding sites. Specifically these approaches
%reveal protein binding sites only in lower resolution, i.e., at an
%interval of hundreds to thousands of base pairs. Furthermore, they
%implicitly assume that there is only one ``mode'' or ``predicted
%binding location'' per this wide genomic interval. More recently,
%deconvolution algorithms such as Deconvolution (Lun et al.,
%2009\cite{csdeconv}), GEM (Guo et al., 2012\nocite{gem}, an improved
%version of Guo et al., 2010\nocite{gps} ) and PICS (Zhang et al.,
%2010\nocite{pics}) have been proposed to identify binding sites in
%higher resolution using ChIP-Seq data. However, most of them are still
%not tailored for ChIP-exo and PE and SE ChIP-Seq data in a unified
%framework and as a result, currently available methods are not
%appropriate for fair comparison between ChIP-exo and ChIP-Seq. To
%address these limitations, we developed and utilized an improved
%version of dPeak (Chung et al., 2013\nocite{dpeak}), a high resolution
%binding site identification (deconvolution) algorithm that we
%previously developed for PE and SE ChIP-Seq data, so that it can also
%handle ChIP-exo data. The dPeak algorithm implements a probabilistic
%model that accurately describes the ChIP-exo and ChIP-Seq data
%generation process.

%Some of the key findings in this work are as follows. First, we
%demonstrate that the ``peak pair'' assumption of Rhee and Pugh,
%2013\nocite{exo2} does not hold well in real ChIP-exo data. Second, we
%found that when analyzing ChIP-exo data and the Input control is not
%available, it is useful to adjust for GC content and mappability
%biases to improve peak calling and binding site identification. Third,
%we evaluated several methods to identify binding events and dPeak
%performs competitively respect to GEM and MACE when analyzing ChIP-exo
%data. Finally, when comparable number of reads is used for both
%ChIP-exo and ChIP-Seq , dPeak coupled with ChIP-exo data provides
%resolution comparable to PE ChIP-Seq and both significantly improve
%the resolution of protein binding identification compared to SE-based
%analysis with any of the available methods.

\section*{Results and discussion}
\label{sec:results}

\subsection*{Publicly available ChIP-exo/nexus and novel \textit{E. coli} ChIP-seq/exo datasets}
We utilized a rich collection of publicly available ChIP-exo/nexus data from multiple organisms to build and evaluate our quality control pipeline (Table~\ref{tab:qc}). These include:   CTCF factor
in human HeLa cells \cite{exo1}; ER factor in human MCF-7
cells \cite{exoillumina}; GR factor in IMR90, K562, and U2OS human
cells \cite{starick15}; TBP factor in human K562 cells
\cite{venters13}. 
ChIP-nexus data included experiments from \cite{chipnexus} profiling
TBP in human K562 cells, MyC and Max in \emph{D. Melanogaster} S2 cells, and Twist and
Dorsal in \emph{D. Melanogaster} embryo.

In order to have a setting where we can compare SE and PE ChIP-seq with their ChIP-exo counterpart, we profiled 
$\sig$ under a variety of  conditions in \textit{E. coli} with both ChIP-exo (Table~\ref{tab:qc_sig}) and PE ChIP-seq (\SK{Table SXX}). Collectively, we 
 generated $\sig$ factor ChIP-exo, PE and SE ChIP-seq experiments  under aerobic ($+O_2$) and anaerobic ($-O_2$)
conditions in glucose minimal media. Similarly, we generated $\sig$ factor ChIP-exo, PE and
SE (generated \emph{in silico}) ChIP-seq experiments in \emph{E. Coli}
under aerobic ($+O_2$) conditions with  and without  rifampicin  treatment.
We refer to these latter set of experiments  as time point 20 minutes and time point 0 minutes \SK{Check for consistency with other parts of the paper and correct, ok to use different naming convention}.
We specifically used these experimental data  for comparisons of ChIP-exo and PE
ChIP-seq assays in identifying closely spaced binding events and their resolution, i.e., physical proximity of the predicted events to the actual binding sites.  This comparison benefits from using $\sig$  which is a transcription initiation factor of housekeeping genes
in \emph{E. Coli}. Many \textit{E. coli} promoters contain
multiple transcription start sites (TSS). These TSSs are often
closely spaced, i.e., within 10 $\sim$ 150 bps of each other, and are considered  to be multiple ``switches'' that differentially
regulate gene expression under diverse growth conditions
\cite{regulondb}.


\subsection*{ChIP-exo versus ChIP-seq: general features}

\textit{Read distributions within signal and background regions.}
We first compared ChIP-seq and ChIP-exo in terms of data features that are well studied in ChIP-seq studies.
Our $\sig$  ChIP-seq and ChIP-exo samples from \textit{E. coli} are especially well suited for this task since they are all deeply sequenced compared to the genome size of \textit{E. coli}.   Figure~\ref{fig:comp} summarizes this comparison for one biological replicate  of ChIP-exo and ChIP-seq experiments from the same biological conditions (samples x and y from Table~\ref{tab:qc_sig} \SK{Maybe add a column as sample number to this table to make referencing easy}). Comparisons with other paired \textit{E. coli} ChIP-seq and ChIP-exo  samples led to similar conclusions (Supplementary Figure \SK{SXXX}).
%compared various factors that could affect binding site
%identification between ChIP-exo and ChIP-Seq data by using the
%ChIP-exo reads from the first biological sample and first replicate
%grown under aerobic condition (first line from Table \ref{tab:qc_sig})
%and a SE ChIP-Seq replicate grown under the same conditions. 
We summarized the extended raw read counts within $150$ bps non-overlapping intervals, i.e., bins, interrogating the genome.
%In order
%to compare distribution of signal and background between ChIP-exo and
%ChIP-Seq data, we counted the number of extended reads mapping to a
%partition of the genome into non-overlapping bins. 
Figure~\ref{fig:comp}(A) depicts that, as expected,  ChIP read counts from ChIP-exo and ChIP-seq are linearly 
correlated especially at high read counts. This indicates that signals for potential binding
sites are well reproducible between ChIP-exo and ChIP-seq data. 
In contrast, there is a clear difference
among the two data types for bins with low read counts, highlighting expected differences in the  background read distributions 
of the two data types. 
\SK{***After seeing these types of plots from other samples (e..g., ER in MCF7, it is unlikely that what we are seeing from $\sig$ samples regarding background reads (the next few sentences) is typical. For CTCF and $\sig$, the ChIP-exo samples have a wider dynamic range, but this could be due to differences in sequencing depth. *** Specifically, background reads from ChIP-seq are  almost
uniformly distributed over background (non-enriched) regions. In comparison, background regions in ChIP-exo show larger variation but have overall lower read counts. Furthermore, there are considerably large numbers of bins with zero read counts in ChIP-exo and non-zero read counts in ChIP-seq.
%and the
%ChIP tag counts in there regions were significantly larger than
%zero. In contrast, in ChIP-exo data, there was larger variation in
%ChIP tag counts among background regions and ChIP tag counts were much
%lower in these regions compared to ChIP-Seq data. There were also
%large proportion of regions without any read in ChIP-exo data. 
Overall, these observations indicate that  a much smaller portion of the
genome is expected to be background in ChIP-exo compared to ChIP-seq and 
 methods that
specifically model the background read distribution might benefit from acknowledging this.}

\textit{Peak-pair assumption.} We next evaluated the peak-pair
assumption, i.e., a cluster of reads
in the forward strand is usually paired with a cluster of reads in the
reverse strand that is located on the right-hand-side of  the binding
site, that is commonly utilized in designing statistical analysis methods for ChIP-exo data \cite{mace, cexor, peakzilla}.
We considered the set of peaks identified in both the ChIP-seq and ChIP-exo samples as high quality peaks (Materials and Methods) and calculated the proportion of forward strand reads in these regions (Figure~\ref{fig:comp}(B)). This plot reveals a higher level of strand imbalance for ChIP-exo compared to ChIP-seq. 
%In order to evaluate this assumption, we reviewed the proportion of reads in the forward strand in high quality ChIP-exo peaks such as at least one binding site is predicted in both ChIP-exo and ChIP-Seq data.
%We
%found that strands of reads were much less balanced in ChIP-exo data
%than in ChIP-Seq data in these regions with potential binding sites
%(Fig. \ref{fig:comp}B) and this indicates that the peak pair
%assumption might not hold in ChIP-exo data. 
Potential reasons for this observation include ligation efficiency, efficiency of $\lambda$-exo digestion, 
and single-stranded protein-DNA interactions. Overall, such an imbalance is prevalent in  \SK{X}\% of the ChIP-exo samples used in this paper.
%the enzyme digestion or the strength at which the protein binds to the
%DNA: Although it is expected for the exonuclease enzyme to digest the
%ChIP fragments starting by their $5\prime$ ends and stopping when
%finding a binding event, it may not be able to reach and digest every
%fragment in the ChIP sample. On the other hand, if the protein is not
%bound to both DNA strand, the enzyme may completely digest the
%fragment. Finally both effect are increased by the PCR amplification.

\textit{Mappability and GC-content bias.} We next evaluated ChIP-exo data of CTCF in HeLa cells
\cite{exo1} to investigate biases inherent to next generation sequencing experiments with eukaryotic genomes. 
Figures \ref{fig:comp}(C) and
\ref{fig:comp}(D) display the bin-level average read counts against
mappability and GC-content. Each data point is obtained by averaging
the read counts across bins with the same mappability of GC-content. 
These biases, increasing linear trend with mappability and non-linear trend with GC-content, are similar to those observed in ChIP-seq datasets \cite{benjamini2011, mosaics,quest}.
%In Figure \ref{fig:comp}C it is shown that the ChIP-exo tag
%counts linearly increases with the mappability score and in Figure
%\ref{fig:comp}D it is shown that for GC - content below 0.6, the mean
%ChIP tag count increases and for GC - content greater than 0.6 it
%shows a decreasing trend. Benjamini and Speed,
%2011\nocite{benjamini2011} and Kuan et al., 2011\nocite{mosaics}
%studied the presence of the mappability and GC - content biases in
%ChIP-Seq's background. It is not surprising to see these biases also
%present in ChIP-exo data, since ChIP-Seq and ChIP-exo signal seems to
%be linearly correlated for enriched regions (Figure
%\ref{fig:comp}A). Rozowsky et al., 2009\nocite{peakseq} and Valouev et
%al., 2008\nocite{quest} provide in depth analysis of the mappability
%and GC - content biases for ChIP-Seq respectively. 
This observation  indicates that analysis of  ChIP-exo data
should benefit from methods that take into account  apparent
sequencing biases such as mappability and GC content, mostly when an
input  control sample is not available.

\subsection*{Application of ENCODE ChIP-seq quality metrics   to ChIP-exo and
  ChIP-nexus data}

ENCODE consortium established empirical and widely used QC metrics on ChIP-seq data
\cite{encode_qc}. Currently, these constitute the state-of-the-art QC pipelines for these high throughput experiments.
We evaluated how these metrics, namely Normalized Strand
Cross-Correlation (NSC), Relative Strand Cross-Correlation (RSC), and PCR
Bottleneck Coefficient (PBC) defined at 
\url{https://genome.ucsc.edu/ENCODE/qualityMetrics.html} \cite{encode_qc}, behave
 on ChIP-exo/nexus data in  Tables~\ref{tab:qc} and \ref{tab:qc_sig}.

%We continued our exploration by investigating whether the current
%state-of-the-art QC pipelines for ChIP-Seq are suitable for ChIP-exo
%and ChIP-nexus. In Tables \ref{tab:qc_sig} and \ref{tab:qc} we
%calculated a collection of the commonly used ChIP-Seq QC metrics using
%the ChIP-exo and ChIP-nexus experiments instead: Normalized Strand
%Cross-Correlation (NSC), Relative Strand Cross-Correlation and PCR
%Bottleneck Coefficient (PBC) defined as in
%\url{https://genome.ucsc.edu/ENCODE/qualityMetrics.html}\nocite{encode_qc}.

\arrayrulewidth=.1mm 


\begin{table}[h!]
  \centering
  \begin{tabu} to\linewidth{X[1.3]|X[2.2]|X[-1]|X[1.3]|X[-1,m,c]|X[1.8]|X[-1]|X[-1]|X[-1]}
    \firsthline
    \textbf{Protocol} & \textbf{Organism} & \textbf{TF} & \textbf{Cell type} & \textbf{Rep.} &
    \textbf{Depth} & \textbf{NSC} & \textbf{RSC} &   \textbf{PBC} \\ 
    \hline
    \multirow{13}{1em}{ChIP-exo} & Human & CTCF & HeLa & & 48,478,450 & 16.02 & 1.1960 & 0.4579 \\
    & \multirow{3}{1em}{Human} & \multirow{3}{1em}{ER} & \multirow{3}{3em}{MCF-7} & 1 & 9,289,835 & 
    19.87 & 1.0127 & 0.8082 \\
    &  &  & & 2 & 11,041,833 & 21.48& 1.0063 & 0.8024\\
    &  &  & & 3 & 12,464,836 & 18.72& 1.0100 & 0.8203 \\    
    &  \multirow{3}{1em}{Mouse} & \multirow{3}{1em}{FoxA1} & \multirow{3}{1em}{Liver} & 1 & 22,210,461 & 21.28 & 1.1104 &  0.6562 \\
    &  & & & 2 & 23,307,557 & 60.42 & 1.1604 & 0.7996 \\ 
    &  & & & 3 & 22,421,72  & 72.04 & 1.1975 & 0.1068 \\
    & \multirow{3}{1em}{Human} & \multirow{3}{1em}{GR} & IMR90 & & 47,443,803 & 8.86  & 1.3678 & 0.2978 \\
    & &  & K562 & & 116,518,000   & 4.11 & 1.0441 &0.0504 \\
    & &  & U2OS & & 3,255,111 &  10.05 & 1.0288 & 0.7714 \\
    & \multirow{3}{1em}{Human} & \multirow{3}{1em}{TBP} & \multirow{3}{1em}{K562} &1 & 61,046,382 &  12.01  & 1.1119
 & 0.1232 \\
    & & & &2 & 94,314,770 & 7.93 & 1.0299 & 0.1681 \\
    & & & & 3 & 114,282,270 & 9.25 & 1.1027 & 0.1464\\
\hline
\multirow{10}{1em}{ChIP-nexus} & \multirow{8}{1em}{D.Melanogaster} &
       \multirow{2}{1em}{Dorsal} & \multirow{4}{1em}{embryo} & 1 &   8,863,170 &  7.27 & 1.0402 &  0.6766 \\
 & & &  & 2 & 10,003,562 & 7.19 & 1.0672 & 0.5656\\
 & &  \multirow{2}{1em}{Twist} & & 1 & 18,244,203 & 5.82 &  1.1637 &  0.6592 \\
 & & &  & 2 & 52,546,982 & 5.27  &  1.1805  & 0.4549 \\
 & & \multirow{2}{1em}{Max} & \multirow{4}{1em}{S2} & 1 & 18,320,743 & 3.60 & 1.3628 & 0.5178  \\
 & & & & 2 & 24,965,642  & 3.47 & 1.0138 & 0.2124\\
 & & \multirow{2}{1em}{MyC} & & 1 & 7,832,034 & 5.92 & 1.0115 &  0.3935  \\
 & & & &  2 & 22,824,467 & 5.76 & 1.0045 & 0.1879\\
 & \multirow{2}{1em}{Human} & \multirow{2}{1em}{TBP} & \multirow{2}{1em}{K562} & 1 &  
              33,708,245 & 32.16 & 1.1712 & 0.3102 \\
 & &  &  & 2 & 129,675,001 & 32.70 &  1.2455 & 0.0492 \\
    \lasthline
  \end{tabu}
  \caption{Summary of publicly available data used for development and evaluation of \pname{}. The last three columns depict ENCODE QC metrics on these data:  NSC:  Normalized 
    Strand Cross-Correlation; RSC: Relative Strand Cross-Correlation;  PBC:  
    PCR Bottleneck Coefficient.}  
\label{tab:qc}
\end{table}

\begin{table}[h!]
  \centering
  \begin{tabu} to\linewidth{X[-1,m,c]|X[-1]|X[-1]|X[-1,m,c]|X[-1]|X[-1]|X[-1]|X[-1]}
    \firsthline
    \textbf{Bio Sample} & \textbf{Condition} & \textbf{Treatment} & \textbf{Rep.} & \textbf{Depth} & \textbf{NSC} & \textbf{RSC} & 
    \textbf{PBC}\\
    \hline 
    \multirow{4}{1em}{1} & Aerobic & No Rif. & 1 & 13,961,493 & 103.15 & 2.0193 & 0.1399 \\
    & Aerobic & No Rif. & 2 & 14,810,838 & 162.70 & 1.7805 & 0.1633 \\
    & Anaerobic & No Rif. & 1 &  16,108,774 & 153.51 & 1.8035 & 0.1353 \\
    & Anaerobic & No Rif. & 2 &  13,636,541 & 172.59 & 2.014 & 0.1532 \\ 
    \hline
    \multirow{4}{1em}{2} & Aerobic & No Rif. & 1 & 902,921 & 13.77 & 1.1270 & 0.2689\\
    & Aerobic & No Rif. & 1 & 1,852,124 & 17.91 & 1.5275 & 0.2590\\
    & Aerobic & Rif. 20 min & 2 & 2,104,427 & 29.60 & 1.2844 & 0.2584\\
    & Aerobic & Rif. 20 min & 2 & 11,548,572 & 13.08 & 1.5122 & 0.1510 \\
    \lasthline
    \end{tabu}
    \caption{ Summary of the \emph{E. Coli} $\sig$ ChIP-exo and ChIP-seq samples. 
The last three columns depict ENCODE QC metrics on these data:  NSC:  Normalized 
    Strand Cross-Correlation; RSC: Relative Strand Cross-Correlation;  PBC:  
    PCR Bottleneck Coefficient.    }
\label{tab:qc_sig}
\end{table}


DNA libraries generated by ChIP-exo and ChIP-nexus protocols are
expected to be less complex than the libraries generated by ChIP-seq because
 the numbers of positions to which the reads can 
align to are reduced  due to the exonuclease digestion. 
%considerable amounts of reads are being mapped to specific positions.
This affects the interpretation of the PBC, which is defined as the ratio of the 
number of genomic positions to which exactly one read maps to   the number of genomic positions to which at least one  read maps.
For ChIP-seq samples,  low PBC values (e.g., $\le 0.5$) indicate high levels of PCR amplification bias, i.e., PC bottleneck, 
 unless the sequencing depth is high enough to sature all targets of the factor profiled.
In contrast,  for ChIP-exo/nexus, exonuclease digestion will lead to reads with same exact 
$5^{\prime}$ end even before the PCR amplification step.
%because of the exonuclease digestion,
% several reads that their $5\prime$ end was digested to
%the same position before the amplification step.
We note  that  the PBC values are especially low
for deeply sequenced ChIP-exo and ChIP-nexus
samples; however, this does not automatically indicate severe 
bottlenecking as suggested by standard ChIP-seq guidelines.
%, which by following the
%ChIP-Seq guidelines it would indicate that those experiment show
%severe bottlenecking problems, which may incorrectly suggest that the
%positions with a large amount of aligned reads are being caused by PCR
%amplification rather than observed genomic signal.

The Strand Cross-Correlation (SCC), introduced by \cite{strandcc}, is the most commonly used quality metric in assessing 
ChIP-seq enrichment quality. It aims to quantify  how well the reads mapped to each
strand are clustered around the locations of the protein-DNA interaction sites by calculating the Pearson correlation among forward and backward strands reads by shifting them across a range that covers both the read length of the experiment and the expected average fragment length.
Typical SCC profiles exhibit two local maxima: at the average fragment length and the read length. In high quality experiments with clear ChIP enrichment, the average fragment length maximum coincides with the global maximum.
%, and usually it is expected to observed two local
%maxima, one when the profiles are shifted by the average read length
%and another when the profiles are shifted by the unobserved fragment
%length. 
%In a high quality ChIP-Seq dataset the last one is also the
%SCC global maxima. 
In an
idealized ChIP-exo experiment where the DNA fragments are digested
 to  the boundaries of the protein-DNA interaction sites,  we would expect the SCC profile to maximize at the motif length indicating clustering of the forward and reverse strand reads around the  binding site.
 Figure~\ref{fig:scc_exo} displays the SCC curves for the CTCF  HeLa samples where the ChIP-exo curve shows  local maxima at the motif and read
lengths, while the SE ChIP-seq curves have a local maxima at the read
length and a global maxima at the average fragment
length. SCC profiles for other samples  are available in Supplementary Figures \SK{XXX}. The read length and motif length maxima are often in close proximity of each other; as a result, this renders
QC metrics such as the Normalized
Strand Cross-Correlation (NSC) or the Relative Strand
Cross-Correlation (RSC) harder to interpret; however, the profile itself seems informative about the enrichment signal in ChIP-exo/nexus experiments \SK{this seems true at least based on my initial look at the plots, but we should be cautious in reporting this}.

\subsection*{ChIP-exo quality control pipeline}

We first present the overall pipeline and then discuss
individual components with a case study using ChIP-exo data of FoxA1  
from \cite{exoillumina} and ChIP-nexus data from \cite{chipnexus}.
Figure~\ref{fig:qcdiagram} summarizes  the 4-step pipeline. 
Given aligned reads from a  ChIP-exo/nexus sample, the first step partitions the reference genome into islands by keeping the
non-digested ChIP-exo regions.  In step 2,   the total
number of extended reads \SK{check: extended or not? What do we extend to in ChIP-exo? } overlapping each island ($D_i$) and the number of
unique island positions with at least one aligned read re recorded ($(U_i)$. Then,  three summary statistics $\text{ARC}_i$, $\text{URCR}_i$, and $\text{FSR}_i$ are computed for each region $i$.  $\text{ARC}_i$ denotes the \textit{average read coverage per base pair} and is defined as the ratio of the
\#  of reads in island $i$ ($D_i$) to the width of the island $i$ ($W_i$);  $\text{URCR}_i$, \textit{unique read coverage ratio}, \SK{Is unique read coverage ratio a good name for this?} quantifies inverse of the effective coverage and is defined as the ratio of the \# of genomic positions with at least one aligned read within island $i$ ($U_i$) to the \# of reads in island $i$ ($D_i$);  and  $\text{FSR}_i$ denotes the proportion of forward strand reads.
%\begin{align*}
%  \text{ARC}_i &= \dfrac{\text{\#  of reads in island $i$}}{\text{Width of the island $i$}}, \\
%  \text{URCR}_i & = \dfrac{\text{\# of genomic positions with at least one aligned read within island $i$}}{\text{\# of reads in island %$i$}}, \\
%  \text{FSR}_i &= \dfrac{\text{\# of forward strand reads in island $i$}}{\text{\# of reads in island $i$}}.
%\end{align*}
Step 3 of the pipeline generates several diagnostic plots aimed at quantifying ChIP enrichment and strand imbalance and step 4 generates quantitative summaries of these diagnostic plots \SK{Aren't the linear model fit etc part of the pipeline?}

Figure~\ref{fig:qcdiagram}A presents
the typical behavior of the URCR vs. ARC plot for a high quality ChIP-exo sample. In general, the plot
depicts two strong arms. 
High URCR values correspond to regions with reads
concentrated on a small number of positions. 
The  left arm, with low ARC  and
varying URCR values,  corresponds to  background islands, regions that
are usually composed of scattered reads that were not digested during
the exonuclease step. The right arm  where the URCR decreases as the
ARC increases  corresponds to regions that are usually ChIP enriched.
%and as the URCR decreases the library complexity does it as well, on
%the other hand 
\SK{Not sure what this next sentence is referring to: Finally we quantify we quantify the
relationship between library complexity by the use of two indexes that
represent the change in the number of unique positions per regions and
the change in of the width of a regions as the depth changes. }
Figures~\ref{fig:qcdiagram}B and \ref{fig:qcdiagram}C aim to quantify the strand imbalance as part of the QC pipeline. The former  depicts how
quickly the islands exclusively formed by reads SK{***I changed fragments to reads, but should this stay as fragments***} from a single strand are  filtered out as islands with higher depths are observed. In a
high quality sample,  the proportion of islands with reads from only one strand 
is expected to decrease rapidly  as we consider higher depth regions. In contrast, 
 this proportion remains
approximately constant in  lower quality samples. The latter plot illustrates how quickly the quantiles of 
the  FSR approaches to 0.5, the expected  FSR value in high quality samples.
%median, since in a high quality sample it is
%expected for the median to be approximately 0.5 and the enriched
%regions are going to be composed by fragments sequenced from both
%strands.

\subsubsection*{Application and validation of \pname{} with the FoxA1
  ChIP-exo dataset}
\SK{We should present the whole QC results and end with validation -  panels from the two figures should be switched.}
We next use Fox A1 ChIP-exo datasets, with three biological replicates at comparable sequencing depths from mouse liver cells, 
to illustrate the QC pipeline. 
Figure~\ref{fig:enrich}A presents  URCR vs. ARC plots
for all three replicates. The first and third replicates exhibit a defined
decreasing trend in  $\mbox{URCR}$ as the $\mbox{ARC}$ increases. This indicates that
these samples exhibit a higher ChIP enrichment than the
second replicate. On the other hand, the overall $\mbox{URCR}$ level
from the first two replicates is higher than that of the third replicate
level,  elucidating  that the  libraries for the first two replicates are more complex than that of  the
third replicate. 
%As we discussed above,  background fragments are often digested by
%the exonuclease enzyme in ChIP-exo experiments; therefore, the balance between the enrichment
%and library complexity of an experiment is one of the key factors determining
%the sample's data quality.
%Using the Fox A1 in mouse liver cell lines generated by Serandour et
%al., 2013\nocite{exoillumina} and these two quantities, we explored
%the relationship between library complexity and experiment
%enrichment. 

%\textbf{Analysis of strand imbalance in FoxA1 ChIP-exo data}

%The strand imbalance assessment is based in the observation that the
%enriched regions usually are composed of a higher quantity of reads,
%therefore we examined the FSR as the regions with lower depth are
%being filtered out. This indicator is of particular importance, as it
%evaluates the ``peak pair'' assumption that the original ChIP-exo
%paper suggested and multiple ChIP-exo data analysis methods rely
%on. For every ChIP-exo experiment, we calculated the global FSR and
%noticed that for all experiments is roughly $0.5$, which means there
%are approximately the same amount of reads in both strands.

Figure~\ref{fig:strand}A and B display the strand-imbalance diagnostic plots and highlight specific problems with replicates 2 and 3. In particular, \SK{I have a hard time interpreting B and C specifically, need to include something about their implications for replicates 2 and 3.}
%In order to study the strand imbalance of the FoxA1 factor ChIP-exo
%experiments in mouse liver generated by Serandour et al.,
%2013\nocite{exoillumina} we considered the visualizations shown in
%Figure \ref{fig:strand}: Figure \ref{fig:strand}A presents the FSR's
%behavior as the lower depth regions are being filtered out, while
%Figure \ref{fig:strand}B) shows the proportion of regions that are
%composed by reads from both strands against the regions formed by
%reads in exclusively one strand. In a high quality dataset, it is
%expected for all quantiles to quickly converging towards the median
%(in panel A) or the regions formed by reads in one strand (either
%forward or backward) to be formed by few fragments (in panel
%B). 
%Additionally, it is worth mentioning that the background in a ChIP-exo
%experiment is expected to be formed by the undigested ChIP fragments
%that are not bound to a protein, hence it is for the reads in both
%strand to be imbalanced in a background region. 
\SK{I also don't follow the discussion for panel C: Figure
\ref{fig:strand}C compares the strand imbalance when ChIP-exo islands 
overlap with high quality peaks. It is noticeable that for regions
composed by a large amount of reads, it is harder to distinguish their
peaks by considering only the strand imbalance, hence in a high
quality ChIP-exo experiment the background is expected to show a
higher imbalance that the enriched regions. In conclusion, Figure
\ref{fig:strand} shows that the global FSR does not accurately
represent a ChIP-exo experiment's strand imbalance locally, hence the
``peak pair'' assumption does not completely hold in every ChIP-exo
enriched region.}


Overall, we conclude that replicate 1 is higher quality than both  of replicates 2 and 3.
We validate this observation with a motif analysis on the candidate binding regions identified from these replicates \SK{Could we say binding events? or is the peak analysis on the whole binding region, i.e., peak? If so, we may want o change "candidate regions" to peaks?} 
Figure~\ref{fig:enrich}B summarizes the total numbers of regions identified from each replicate. The lower number of enriched regions from replicate 2 is consistent with the lower ChIP enrichment pattern in the $UCRC$ vs. $ARC$ diagnostic plot.  Scanning of these regions for the occurrence of FoxA1 sequence motif with the FIMO tool \cite{fimo} indicates that the first replicate outperforms the other two in terms of percentage of candidate regions with the FoxA1 motif (Figure~\ref{fig:enrich}C).
%Considering the regions and summary statistics we obtained a set of
%candidate sites for each replicate, the total number of candidate
%sites is shown in Figure \ref{fig:enrich}B.
%For this sites, we
%extracted the sequence and searched for the FoxA1 motif using
%\textbf{FIMO} \cite{fimo}. Figure \ref{fig:enrich}C shows the fraction
%of candidate site where a motif was found. The first replicate
%outperforms the rest in both number of candidate sites and proportion
%of sites with motif; on the other hand we can see that the second
%replicate candidate sites are more likely to contain the motif
%sequence, but there more candidate sites in the third replicate than
%in the second. 
Furthermore, Figure~\ref{fig:enrich}D displays the average normalized
read coverage around the actual motif locations in the candidate binding regions.
These coverage plots reveal that the ChIP signal is more defined for
the first and third replicates than the second one, indicating overall strength of the ChIP enrichment in these samples compared to the second replicate.
% however,  overall, the signal pattern is reproducible in all of the three biological replicates.
Figures \ref{fig:enrich}E, \ref{fig:enrich}F, and
\ref{fig:enrich}G further highlight the overall quality of the identified motif sequences for each replicate and suggest  that libraries with high library complexity (replicates 1 and 3) capture binding sites with better motif matches.
\SK{Worth thinking a bit more about the interpretation.}

%In summary,  the diagnostics plots of the QC pipeline highlights that  observing a defined
%decreasing trend in $\mbox{URCR}$ as $\mbox{ARC}$ increases implies
%ChIP enrichment and multiple samples can be compared with respect to the degree of this trend. 
%Second, for experiments that are not deeply
%sequenced, low $\mbox{URCR}$ values represent low complexity. 
%decreases the amount of sites of motif found but it doesn't modify the
%read distribution around the motif. Finally, libraries with higher
%complexity returns sequences with a better motif match. \SK{We do not have enough analysis with motif validation to conclude these.}

\subsubsection*{High sequencing depth may confound low-complexity library issues}
We next evaluated every sample listed in Tables~\ref{tab:qc} and
\ref{tab:qc_sig} with the \pname{} QC pipeline (Supplementary Figures \SK{XXX}). A key, albeit not suprising, observation from large scale analysis is that  the URCR vs. ARC plots  typically display the three patterns captured in the FoxA1 study. We will refer to these as pattern I (FoxA1 replicate 1), II (FoxA1 replicate 2), and III (FoxA1 replicate 3), respectively. 
Pattern III where the two arms along $\mbox{ARC}$ are not distinguishable can arise due to either low-complexity library or high sequencing depth.  For example, all three replicates of the TBP ChIP-exo from K562, with sequencing depths between $\sim$ 60M to 115M reads, and replicate two of TBP ChIP-nexus in K562, with sequencing depth of $\sim$ 130M reads, exhibit this pattern. A simple but effective strategy to distinguish the two plausible scenarios from Pattern III is to apply the QC pipeline to sub-samples randomly generated from the full dataset at varying sequencing depths.  We applied this strategy by  sub-sampling 20M to 50M reads, a range that represents the 
 sequencing depths of the human samples we are using in the paper, from the TBP samples. 
$\mbox{URCR}$ vs. $\mbox{ARC}$ diagnostics of these sub-samples  (Supplementary Figures \SK{XXX}) indicate that of the four TBP samples with this pattern, replicates two and three of K562 ChIP-exo suffer from low-complexity library issues, whereas the other samples exhibit the pattern specific to high quality samples. \SK{Add: Validation with motif on TBP ChIP-exo replicates.}

  
\subsubsection*{Evaluation of a large collection of ChIP-exo and ChIP-nexus data with \pname{}}

We next performed an overall analysis of the \pname{} 
QC pipeline results for the samples in Tables~\ref{tab:qc} and
\ref{tab:qc_sig}.  We  quantified the relationship between $\mbox{ARC}$ and $\mbox{URCR}$ by fitting a reparametrized regression model of $\mbox{URCR}$ as a function of  $\mbox{ARC}$.   Specifically, we considered $D_i = \beta_1 U_i +\beta_2 W_i +\varepsilon_i$, where 
$\varepsilon_i$ represents the random error term.  As we discuss in Materials and methods, this parametrization has a direct connection to $\mbox{URCR}_i = \frac{\kappa}{\mbox{ARC}_i} + \gamma+ \epsilon_i$,   which aims to recapitulate the relationship in the  $\mbox{URCR}$ vs. $\mbox{ARC}$ plots.
%between depth, width, and the number of unique
%positions to summarize the shape obtained in the $\mbox{ARC}$
%vs. $\mbox{URCR}$ visualization. 
Figure~\ref{fig:eval}A displays estimated
overall change in depth ($\hat{\beta}_1$) as the number of positions with at least one aligned read varies across a large collection of ChIP-exo samples from eukaryotic genomes. The $\beta_1$ parameter can be interpreted as the  limiting (i.e., large depth) 
\SK{need a better name for:} average read depth intensity $\mbox{URCR}$ of a sample.
%This parameter can be
%interpreted as the inverse of $\gamma$ in equation (\ref{mod}), which
%can is also the experiment's large depth $\mbox{URCR}$. 
As discussed earlier, high
quality ChIP-exo samples are expected to have two arms in the $\mbox{URCR}$
vs. $\mbox{ARC}$ plots: one with low $\mbox{ARC}$
and varying $\mbox{URCR}$ and another with a decreasing
$\mbox{URCR}$as  $\mbox{ARC}$ increases and stabilizes
$\beta_1$. When the ChIP-exo sample is not deeply sequenced, high
values of $\hat{\beta}_1$ in  Figure~\ref{fig:eval}A indicate that the library complexity
is low. On the other hand, lower values correspond
to higher quality ChIP-exo experiments. Taking into account the depths of these samples and visualizing all the diagnostic plots (\SK{Supplementary Figures XXX}), we conclude that samples with estimated $\beta_1$ values less than $10$ seem to be high quality samples.


We interpret the $\beta_2$ parameter above as the bias for the average read coverage  and display its estimates across all the eukaryotic samples in Figure~\ref{fig:eval}B.
% shows the average read coverage bias estimates when the
%experiment's sequencing depth is large/
 Under perfect digestion by $\lambda$-exo, most of the reads aligned to binding regions are expected to accumulate
 around a binding event.  
 \SK{I don't follow the next argument: We should say something about the expected behavior of $\beta_2$
 which suggest to be unlikely to
observe as the sequencing depth increases to observe a reads being
aligned to another position that was not previously covered. Low
quality ChIP-exo experiment exhibit more scattered reads in both
strands across the genome; therefore it is more likely to observe
consider reads that align to position in the genome that were not
previously covered. }
Although the 
third replicate of the TBP ChIP-exo experiment has comparable sequencing depth to the 
second replicate of the TBP ChIP-nexus experiment, the  
(Figure~\ref{fig:eval}(B)), $\mbox{ARC}$ bias is considerably higher for the ChIP-exo experiment.
This potentially indicates that additional sequencing reads in comparison to replicates 1 and 2 are scattered around new positions instead of accumulating on the existing binding sites.
%We can compare the second replicate from the human
%ChIP-nexus experiment against both second and third replicates from
%the TBP factor experiment in K562 cell lines. The sequencing depths of
%these three experiments  are comparable but the large depth $\mbox{ARC}$'s
%bias is considerably higher for both ChIP-exo experiments.


The interaction between these two parameters  has implications regarding the quality of 
a ChIP-exo and ChIP-nexus sample. When either the adjusted
$\mbox{ARC}$ or the $\mbox{ARC}$ bias is large \SK{Not clear what large for $\beta_2$ means if we are only looking at a single sample} owing to potentially the high sequencing depth of the sample, we suggest
randomly sub-sampling reads to form samples of lower depth and evaluating the 
sub-samples with the QC pipeline. 
As an illustration, we revisit this strategy for  the three replicates of TBP ChIP-exo in K562 \cite{venters13} and 
second replicate from the K562  ChIP-nexus experiments \cite{chipnexus}.
Figure~\ref{fig:eval}C exhibits an increasing trend in the
estimated $\beta_1$ across varying sequencing depths for replicates 2 and 3 which we deem as lower quality than replicate 1. Furthermore, the estimates with lower depths are still higher than that of the replicate 1 and the overall trend of the ChIP-nexus sub-samples. 
%, where the ChIP-nexus
%trend being considerable lower than both ChIP-exo replicates. 
Figure \ref{fig:eval}D illustrates that the $ARC$ bias remains approximately
constant in ChIP-nexus sub-samples and sub-samples of first replicate of ChIP-exo, while it increases for the second and third
ChIP-exo replicates. This suggests that these two ChIP-exo replicates have
low library complexity and overall lower quality than the ChIP-nexus
samples, regardless of the fact that all three experiments are
deeply sequenced with more than 90M  reads each. Furthermore, the \pname{}  diagnostic plots for each sub-sample (\SK{Supplementary Figures XXX}) illustrate that the two arms of the ARC versus UCRC plots are clearly visible in moderate depth sub-samples of TBP ChIP-nexus data.  \SK{Some more work/reorganization not to have too much repetition with this last paragraph and the section that follows right after FoxA1.}

\subsection*{High-resolution binding event identification with statistical analysis  of
  ChIP-exo data}

Our QC pipeline \pname{} operates on aligned read files and does not require any statistical analysis of the data such as identification of potential binding regions/events. This enables its broad and easy usability before any statistical analysis  for identifying binding events from ChIP-exo/nexus data.  We next evaluated recent analytical approaches for ChIP-exo data analysis and further compared ChIP-exo with PE ChIP-seq data on our \textit{E. coli} samples.


\subsubsection*{Evaluation of available methods in discovering closely
  spaced binding events from \textit{E. coli} ChIP-exo  data}

We specifically considered recently developed Peakzilla \cite{peakzilla}, MACE
\cite{mace}, GEM \cite{gem}, and dPeak \cite{dpeak} in our evaluations with 
high quality \emph{E.Coli} samples (samples \SK{x and y} from the aerobic condition). Among these methods, Peakzilla and MACE are specifically developed for ChIP-exo analysis, whereas both dPeak and GEM can identify high resolution binding events from ChIP experiments, rendering them suitable for ChIP-exo analysis as well. 
%After we determined which biases could affect a ChIP-exo experiment's
%quality we decided to proceed and examine binding sites in high
%resolution on high quality ChIP-exo experiments. We examined first
%both aerobic replicates from the \emph{E.Coli} first biosample.

Figures~\ref{fig:methods_comp}A and \ref{fig:methods_comp}B compare
binding events identified with all the four methods  in terms of resolution,
where the resolution is defined as the distance from a RegulonDB \cite{regulondb}
annotation to its closest prediction. The resolutions of  all the methods
are comparable for the first replicate (Figure~\ref{fig:methods_comp}(A)) and dPeak, on average, has slightly better resolution than the rest in the second replicate (Figure~\ref{fig:methods_comp}(B)). 

dPeak model  has two parameters that further elucidate characteristics of ChIP-exo data.
The $\delta$ parameter of dPeak quantifies the average distance from the
$5^{\prime}$ ends of the reads to the binding event they are profiling and the $\sigma$
parameter is a measure of dispersion of $5^{\prime}$ ends around the binding events.
Figures~\ref{fig:methods_comp}C and
\ref{fig:methods_comp}D compare the densities of these parameters
estimated by dPeak for \textit{E. coli}  ChIP-exo and SE ChIP-seq data. As expected, 
both   are lower for ChIP-exo than for SE
ChIP-Seq, indicating that dPeak, although originally developed for high-resolution binding event identification from ChIP-seq, can learn this information.
%Therefore, we can asses that dPeak is accurately
%representing that the ChIP-exo fragments are allocated more tightly
%around the transcription factor binding sites.

\subsubsection*{Saturation analysis with ChIP-exo and ChIP-seq: Systematic comparison of ChIP-seq vs ChIP-exo under
  varying sequencing depths}
Establishing competitive performance of dPeak compared to other ChIP-exo analysis methods enable comparison of ChIP-exo and ChIP-seq with a unified analysis framework using dPeak. Previous comparisons of ChIP-exo and SE ChIP-seq
were all performed without controlling for the sequencing depths of the samples. More importantly, although it is well established that PE ChIP-seq leads to better resolution that SE ChIP-seq in terms of binding site identification \SK{Cite Qi's BMC Bioinformatics paper}, previous work does not have an in depth comparison of ChIP-exo and PE ChIP-seq. 
%Previously, ChIP-exo and SE ChIP-Seq have been compared only at a
%fixed depth level in the literature, while they did not include PE
%ChIP-Seq as well either. 
In order to address these limitations of the
previous studies, we performed a sub-sampling experiment by sampling fixed numbers of reads from each of the
$\sig$ ChIP-exo, PE ChIP-seq, and SE ChIP-seq datasets. Specifically, for every $N$ reads in ChIP-exo and SE ChIP-seq, $N/2$ read pairs 
were sampled for PE ChIP-seq to operate under fixed sequencing costs.
%Additionally, it is worth noticing that for PE ChIP-Seq we
%sampled both ends of the fragment, hence for each sequencing depth we
%are sampling the half amount of pairs for PE ChIP-Seq than for
%ChIP-exo or SE ChIP-Seq.

Figure~\ref{fig:design} summarizes the comparisons of the three data types for $\sig$ under aerobic condition as a function of sequencing depth. For these computational experiments, we used $\sig$ RegulonDB \cite{regulondb} binding events as gold standard.
Figure~\ref{fig:design}A displays the number of candidate regions (i.e., peaks)
 where at least one  binding event was identified whereas Figure~\ref{fig:design}B depicts the number of identified binding events, i.e., each candidate region can harbor multiple binding events.
In terms of the number of  binding events, ChIP-exo ranks on the conservative side.  
In Figure~\ref{fig:design}C,  we
display the number of correctly identified binding events, where we consider a  RegulonDB event as
correctly identified if an estimated binding event is identified within 15 bps
of it. Finally, Figure~\ref{fig:design}D presents the
resolution defined as the distance from a RegulonDB binding event to the
closest dPeak prediction. These comparisons indicate that although PE ChIP-seq performs much better than SE ChIP-seq in terms of identifying closely spaced binding events, ChIP-exo outperforms PE ChIP-seq at comparable depths.
%It is remarkable that even when the number
%of candidate peaks or the number of predicted events is lower for
%ChIP-exo, it outperforms both PE and SE ChIP-Seq in number of
%identified targets and resolution.

%This may suggest that with ChIP-exo less false positive peaks are
%being called and that when the targets are being identified, dPeak
%estimates binding locations closer to the true location. 
%\SK{The lines do not cross each other, hard to support the following statement:}
%Additionally, we can see that as the read depth increases, all four indicator seem to stabilize and hit a plateau earlier than the cases for ChIP-Seq, which may indicate that with ChIP-exo a smaller amount of reads is needed to identify the same number of targets than ChIP-Seq, but it may be also possible that this is an artifact occurring due to ChIP-exo's lower library complexity. 


%\subsubsection*{Comparison with ChIP-Seq data using dPeak}

Figure~\ref{fig:reso_all} shows comparisons among dPeak analysis of full ChIP-exo, PE
ChIP-seq and SE ChIP-seq $\sig$ datasets \SK{Again, make sure these ChIP-seq samples are introduced before and numbering system will help to refer each sample}. \SK{Are the sequencing depths comparable? We are making a big deal about matching depths and then jumping into this analysis. The only way these results would work is that either the depts are comparable or all the samples have depths sufficient enough for saturation}. Utilizing  RegulonDB binding events as ground truth, 
%We considered the RegulonDB data as ground
%truth, since those are the most recent annotation on \emph{E. Coli}. A
%RegulonDB annotation (Salgado et al, 2012\nocite{regulondb}) was
%considered to be identified if the distance from the closest dPeak
%binding site estimate was less that or equal to 20 bp. 
we computed sensitivity  as the proportion of RegulonDB events 
identified by dPeak in each analysis and the resolution as the minimum
distance between a RegulonDB event and the closest dPeak binding
event prediction. Figure \ref{fig:reso_all}A illustrates that as the mean distance between binding events increases, sensitivity
for all data types increase. Consistent with the sub-sampling experiments, both PE ChIP-seq and ChIP-exo significantly outperform SE ChIP-seq and ChIP-exo exhibits more power than PE ChIP-seq in deconvolving binding events.
%shows that the sensitivity
%increases as the mean distance between binding events increases. When
%the binding events in a peak are closer to each other, both ChIP-exo
%and PE ChIP-Seq are comparable, as the distance increases ChIP-exo
%identifies a higher proportion of the RegulonDB annotations;
%additionally SE ChIP-Seq is significantly less sensitive than both
%ChIP-exo and PE ChIP-Seq. 
Figure~\ref{fig:reso_all}B highlights that 
ChIP-exo and PE ChIP-seq are comparable in resolution with these deeply sequenced samples, while both
protocols significantly outperform SE ChIP-seq.

\section*{Conclusions}
\label{sec:conc}

We presented a systematic exploration of several ChIP-exo/nexus experiments. We
provided a list of factors that reflect the quality of a ChIP-exo
experiment and developed a QC pipeline, named \pname{}. \pname{} takes as input aligned reads and automatically generates several diagnostic plots and summary measures that enable assessing enrichment and library complexity.
%Additionally, a set of
%diagnostics was established to assess the quality of a ChIP-exo
%experiment. 
Our analysis of several datasets indicated that
while the QC pipeline only requires a set of aligned reads
to give a global overview of the quality of a given ChIP-exo dataset, 
implications of the diagnostic plots and the summary measures align well
with more elaborate analysis that is computationally more
expensive to perform and/or requires additional inputs that may not be
available, such as motif occurrences in a set of high quality regions or
resolution analysis based on a gold-standard.

%We studied the shared biases between ChIP-exo and ChIP-Seq data, and
%noticed that for eukaryotic genomes the relationship between ChIP-Seq
%data and either the mappability or the GC content scores are still
%present in ChIP-exo. We also examined ChIP-exo's background and
%noticed that is significantly different from the ChIP-Seq one, since
%it consists of only a small quantity of fragments that was not
%digested by the exonuclease enzyme. Additionally, we showed that we
%have unbalanced number of reads in forward and reverse strands, and
%that in a lower quality ChIP-exo experiment those regions are going to
%be harder to differentiate from the possibly enriched regions.

To the best of our knowledge, we also provide the first systematic comparison 
between ChIP-exo and PE ChIP-seq datasets using our \textit{E. coli} $\sig$ samples. 
This comparison revealed that PE ChIP-seq compares much more competitively with ChIP-exo compared to SE ChIP-seq. However,
overall, ChIP-exo provides the best performance in terms of deconvolving closely spaced binding events and resolution.
The pname{} is available at \url{https://github.com/keleslab/ChIPexoQual}.
%Using a set of annotations as
%gold-standard, we showed that both protocols are comparable in
%resolution and that for regions with more than one binding site,
%ChIP-exo is more sensitive than both SE and PE ChIP-Seq. We made a
%rigorous comparison between fixed depth ChIP-exo, PE ChIP-Seq and SE
%ChIP-Seq, and we probed that for sufficiently complex libraries,
%ChIP-exo experiments can outperform PE and SE ChIP-Seq in number of
%identified targets and resolution. The proposed ChIP-exo QC pipeline
%provides a rigorous, easily interpretable, computationally efficient
%framework to diagnose if the library complexity of a ChIP-exo
%experiment is adequate.

\section*{Materials and methods}
\label{sec:methods}

\subsection*{ChIP-seq/exo/nexus datasets}
% To asses the validity of the ChIP-exo QC pipeline several experiments
% were considered.

\subsubsection*{\textit{E. coli} ChIP-exo and ChIp-seq samples}

We generated ChIP-exo and PE ChIP-seq samples from $\sig$ in
\emph{E.Coli}. For each PE ChIP-seq experiment, an \emph{in silico} SE
version was obtained by randomly sampling one of the two ends in a read pair.

{\color{red}\emph{need to add the growth conditions. \SK{Add something based on dpeak paper, we can ask Jeff to check.}}}

\subsubsection*{Processing of the ChIP-exo and ChIP-nexus samples}
%We gathered a collection of ChIP-exo experiments spanning different
%cell lines and transcription factors: CTCF in HeLA cells \cite{exo1},
%ER in MCF-7 cells \cite{exoillumina}, TBP in K562 cells
%\cite{venters13}, GR in IMR90, K562 and U2OS in K562 cells
%\cite{starick15} and FoxA1 in mouse liver cells
%\cite{exoillumina}. Additionally we used the ChIP-nexus experiments
%generated for He et al., 2015: TBP in human K562, MyC and Max factors
%in S2 cell lines in \emph{D. Melanogaster}; Twist and Dorsal factors
%in \emph{D. Melanogaster} embryo cell lines.
We aligned the read files of the samples listed in Table~\ref{tab:qc_sig}
either following the directions in their original publications when available or with \texttt{bowtie} (version 1.1.2) \cite{bowtie}. 
\SK{include bowtie comment to show how parameters were set.}
%The read files were aligned following the instructions provided in
%their respective sources when available. Otherwise we used
%\emph{bowtie} in its 1.1.2 version.

\subsection*{Generation of a set of high signal regions from \textit{E. coli} samples to assess strand imbalance}

%We used the $\sig$ sample grown under aerobic condition for ChIP-exo and
%SE ChIP-Seq. 

We partitioned the \emph{E.Coli} genome into
non-overlaping intervals, i.e., bins, of length 150 bps and counted the number of reads overlapping each bin. As is usually the practice with ChIP-seq analysis, each read was extended to the average fragment length of $150$ bps towards the $3^{\prime}$ direction.
%for each bin we counted how
%many ChIP-exo and SE ChIP-Seq extended reads overlapped each bin. The
%reads were extended by 150 bp to their $3\prime$ directions.
To evaluate the strand imbalance, we identified a set of high quality peaks
for ChIP-exo and SE ChIP-seq by analyzing both with \texttt{mosaics} \cite{mosaics}
under the  $\mbox{GC content + Mappability}$ and $\mbox{Input only}$ modes for background estimation, respectively.
The subset of these peaks for which \texttt{dPeak} analysis  identified one or more binding events were used in FSR assessments (Figure~\SK{Fig num}).  \SK{Do the plots change if we consider only peaks with one binding event? One could argue that when there are multiple events, there might be more background reads, which could skew the FSR distribution.}
%(with $\mbox{GC content + Mappability}$
%and $\mbox{Input only}$ models respectively), such that each peak
%contained at least one binding event (found by using \textbf{dPeak}
%with at most 5 binding events in each peak). 

%For each peak, we
%calculated the FSR as:

%\begin{align*}
%    \mbox{FSR} &= \frac{\text{Nr. of fwd. strand reads mapped to the region}}                 {\text{Total nr. of reads mapped to the region}} 
%\end{align*}

%We estimated the $\mbox{FSR}$'s densities by using R's \emph{density}
%function using its default parameters, which computes gaussian kernel
%density estimates.

\subsection*{ENCODE  ChIP-seq QC metric guidelines}

% scc and related
% change table to have pbc at the end

We used the ChIP-seq QC metric definitions  established by \cite{encode_qc} and described in detail at
%Landt et al., 2012 which are also listed in
\url{https://genome.ucsc.edu/ENCODE/qualityMetrics.html}.
These QC metrics were calculated with the \textbf{ChIPUtils}
package (version 0.99.0 from 
\url{https://github.com/welch16/ChIPUtils}). 
%This package provides an
%easy to use interface to calculate basic quality control metrics and
%diagnostic plots for ChIP-Seq data.
Empirical data from the ENCODE project suggests the following guidelines for interpretation of the QC metrics for human and mouse genomes: a PBC value
between $0$ to $0.5$  indicates severe bottlenecking,  $0.5$ to $0.8$
moderate bottlenecking,  $0.8$ to $0.9$  mild
bottlenecking, and  $0.9$ - $1$ no bottlenecking.



\SK{I am in favor of skipping these definitions of ENCODE metrics  for the paper - you might want to keep it for your dissertation-}

%\subsubsection*{Strand Cross-Correlation.}

%The strand cross-correlation was proposed by Kharchenko et al.,
%2008\nocite{strandcc} and it may be one of the most used of the
%ChIP-Seq QC metrics. The SCC curve is defined as:

%\begin{align}
%  y(\delta) = \sum_c w_c r\left[ n_c^+ \left(x + \frac{\delta}{2}
%    \right), n_c^- \left( x- \frac{\delta}{2} \right)\right],
%\label{scc}
%\end{align}

%where $y(\delta)$ is the SCC for a strand shift $\delta$, $r$ is the
%Pearson correlation, $w_c$ is the proportion of reads mapped to
%chromosome $c$ and $n_c^S$ is the read count vector for strand $S$ and
%chromosome $c$. The following QC metrics are used to summarize the shape of the SCC curve:

%\begin{align}
%  \mbox{NSC} &= \frac{\max_\delta y(\delta)}{\min_\delta y(\delta)}, \label{nsc} \\
%  \mbox{RSC} &= \frac{\max_\delta y(\delta) -
%    y_{\text{bgd}}}{y_{\text{rl}} - y_{\text{bgd}}}. \label{rsc}
%\end{align}

%where $y_{\text{bgd}}$ is estimated as the background SCC level is
%defined as $y_{\text{bgd}} = \min_\delta y(\delta)$ and
%$y_{\text{rl}}$ is the value of the SCC curve when the shift equals
%the experiment's read length. It is worth noticing that in Landt et
%al., 2012 the RSC is defined as:

%\begin{align*}
%  \mbox{RSC} &= \frac{\max_\delta y(\delta)}{y_{\text{rl}}}.  
%\end{align*}

%In ChIP-exo data, both definitions are equivalent since in a typical
%ChIP-exo experiment $y_{\text{bgd}}$ is approximately zero.

%\subsubsection*{PCR Bottleneck Coefficient.}

%The PCR Bottleneck coefficient is a measure of library complexity in
%ChIP-Seq data:

%\begin{align}
%  \mbox{PBC} = \frac{\text{Nr. of positions to which exactly one
%      unique mapping read is aligned}}{\text{Nr. of positions to
%      which at least one unique mapping read is aligned}} \nonumber
%\end{align}



\subsection*{ChIP-exo quality control with \texttt{R} package \texttt{ChIPexoQual}}

We implemented our proposed QC pipeline with an \texttt{R} package named \texttt{ChIPexoQual}, available at 
\url{https://github.com/welch16/ChIPexoqual}. The analysis in this paper used version 1.0 of the \pname{} package.

\begin{framed}
\noindent \texttt{ChIPexoQual}: 
The package takes in as input a set of aligned reads from a
ChIP-exo (or ChIP-nexus) experiment and performs the following steps.


\begin{enumerate}
\item Identify read islands, i.e., overlapping clusters of reads separated by gaps, from  read coverage. 
  
\item Compute $D_i$,  number of reads in the  island $i$ , and $U_i$, number of island $i$ positions with at least one aligning read, $i=1, \cdots, I$.

\item For each island $i$, $i=1, \cdots, I$, compute island statistics:
  \begin{align*}
    \mbox{ARC}_i &= \frac{D_i}{W_i}, \quad \mbox{URCR}_i = \frac{U_i}{D_i},  \\
    %\mbox{URCR}_i &= \frac{U_i}{D_i}, \\
    \mbox{FSR}_i &= (\text{\# of forward strand reads aligning to island $i$})/D_i, 
  \end{align*}
  where $W_i$ denotes the width of island $i$.
\item Generate diagnostic plots (i) URCR vs. ARC plot; (ii) \SK{Name this FSR plot}; (iii) \SK{Name this FSR plot} as in Figure~\ref{fig:qcdiagram}.
\item Randomly sample $M$ (at least \SK{?}) islands and fit, 
  \begin{align*}
    D_i = \beta_1 U_i + \beta_2 W_i + \varepsilon,
  \end{align*}
  where $\varepsilon$ denotes the independent error term.  Repeat this process $B$ times and generate  box plots of estimated $\beta_1$ and $\beta_2$. 
\end{enumerate}

\end{framed}

%We implemented this ChIP exo QC pipeline in the \emph{ChIPexoQual} R
%package, additionally it generates visualizations as seen in Figure
%\ref{fig:qcdiagram}. We used its 1.0 version which is available in
%\url{https://github.com/welch16/ChIPexoqual}.

\textit{Interpretation of the linear model in the QC pipeline}
The linear model
\begin{align*}
  D_i = \beta_1 U_i + \beta_2 W_i+\varepsilon_i
\end{align*}
 re-parametrization of the following relationship from $UCRC$ vs. $ARC$ diagnostic plot:
\begin{align}
  \mbox{URCR}_i = \frac{\kappa}{\mbox{ARC}_i} + \gamma + \epsilon_i
\label{mod}
\end{align}
with $\beta_1 = 1 / \gamma$ and $\beta_2 = - \kappa / \gamma$. In this
setting, $\gamma$ can be considered as the large-depth
$\mbox{URCR}_i$, i.e., the limiting ratio between the number of positions with at least one mapping read
and depth as the depth tends to infinity. 

\SK{***$\beta_2$ needs more clarification. }
On the other hand, to
interpret $\beta_2 = - \kappa / \gamma $, by expressing $\kappa$ as a
function of $\mbox{ARC}$ and $\mbox{URCR}$ and assuming that $\gamma$
is already estimated, we can observe the following identities:

\begin{align*}
  \kappa &= \frac{\mbox{npos}}{\mbox{width}} - \gamma \mbox{ARC} \\
  \frac{\kappa}{\gamma} &= \frac{1}{\gamma} \frac{\mbox{npos}}{\mbox{width}} - \mbox{ARC} 
\end{align*}

This is important: $\gamma$ approximates the $\mbox{URCR}$ as the
sequencing depth increases, which implies that $- \kappa / \gamma$ can
be interpreted as the large sequencing depth bias of the $\mbox{ARC}$
since as the depth increases, the first term of $\kappa / \gamma$ is
going to approximate the average read coverage:

\begin{align*}
  \frac{\mbox{npos}}{\mbox{width}} \times \lim_{d \rightarrow
    \infty} \frac{d }{\mbox{npos}} \sim \mbox{ARC} 
\end{align*}

Therefore, $\beta_2 = - \kappa / \gamma$ is interpreted as the
$\mbox{ARC}$ bias as the sequencing depth increases.
\SK{***}

\subsection*{Motif analysis of FoxA1 enriched regions}
For each FoxA1 ChIP-exo replicate, we used the ChIP-exo QC pipeline to partition the
mouse genome into a set of islands  with their respective summary
statistics. We  then filtered them into collections of high quality regions
by (i) removing the islands with reads residing only on one strand; (ii) removing the islands with $U_i \le 15$; (iii) removing islands with $D_i < 100$. These thresholds were empirically selected and the overall conclusions were robust to their variation.
%\begin{enumerate}
%\item Removing the regions formed by reads on only one strand.
%\item Removing regions with reads aligned to at most 15 unique
%  positions.
%\item Removing regions with less that 100 reads.
%\end{enumerate}
We used \texttt{FIMO} (version 4.9.1, SK{include command line to show parameters}) \cite{fimo} to identify
the FoxA1 motif within each enriched region using FoxA1 position weight matrix MA0148.1 from the JASPAR database \cite{jaspar}.


\subsection*{Testing for strand imbalance in FoxA1 ChIP-exo replicates}
\SK{ This paragraph could benefit from a bit more polishing.}

We used the ChIP-exo QC pipeline to partition the
mouse genome into a set of islands with their respective summary
statistics. We filtered the islands with reads in only one strand and
 transformed the FSR into an \emph{imbalance index} that is zero
when the resulting region is perfectly balanced in terms of forward and reverse strands reads and infinity when it consists of
reads in one strand exclusively:
\begin{align*}
  \mbox{Imbalance index} = -\log_{10} (4 \times \mbox{FSR} \times (1 -
  \mbox{FSR}))
\end{align*}
We divided the ChIP-exo regions into two classes depending on 
whether or not they overlap  ChIP-exo peaks, where the we identified the peaks with \texttt{mosaics}
using the  $\mbox{GC content} + \mbox{Mappability}$ background model \cite{mosaics} (version 2.9.7) and at a false discovery rate (FDR) level of $0.05$ and minimum read threshold of $100$. 
The  bin size and fragment length for MOSAiCS runs were set to $200$ bps. 
%We called peaks with an FDR of $5\%$, a threshold of
%100 and a maximum gap size of 200 bp. 
We further filtered the resulting peaks by
keeping only the peaks with an average extended ChIP read count of 200 and  used a Wilcoxon test to assess the difference in imbalance index between the two groups of ChIP-exo regions.

%\SK{Is this showing the imbalance index is different between peak and non-peak regions?
%To show that the class that don't overlap with peaks exhibits heavier
%tails, we used a Wilcoxon test over the \emph{Imbalance index}.}

%\subsubsection*{GC - content and Mappability}

%To define the mappability score we follow the definition from Rozowsky
%et al., 2009:

%\begin{align*}
%  m_i &= \sum_{k = i - L +1}^{i + L - 1} \frac{\delta_k }{2L - 1}.
%\end{align*}

%where $\delta_i$ is the indicator if the base at coordinate $i$ can be
%mapped uniquely by a 32 bp sequence at position $i$, and $L$ is the
%expected fragment length. GC - content score is defined analogously,
%where $\delta_i$ represents the occurrence of a G or C at the $i$-th
%position in the genome. 

%The mappability and GC - content scores for a bin are defined as the
%average of the scores across the nucleotides in the bin.

% Hence, when we reject the hypothesis of the
% distribution being the same for both classes, it means that the class
% of region that don't overlap peaks shows heavier tails than its
% complement. \footnote{not sure if I should add the FSR densities for
%   both classes in the supplement to prove this statement}

% \subsection*{Construction of a SE ChIP-Seq from a PE ChIP-Seq experiment.}

% For the rif-treatment ChIP-Seq experiments, we sampled SE ChIP-Seq
% experiment from the PE ones by taking one of both ends randomly
% following with equal probability.

\subsection*{High resolution analysis with ChIP-exo}

In all the evaluations using \textit{E. coli} samples, 
we considered RegulonDB \cite{regulondb} $\sig$ sites as gold-standard. A $sig$ site was deemed 
as identified  if there existed a binding event within its  20 bps proximity. We defined the
resolution as the distance from an $\sig$ site to its closest
predicted binding event and the sensitivity as the fraction of correctly identified
$\sig$ sites in a genomic region.

\subsubsection*{Statistical methods for ChIP-exo}

We compared dPeak \cite{dpeak} (\url{https://github.com/dongjunchung/dpeak}, version 2.0.1), GEM \cite{gem} (\url{http://groups.csail.mit.edu/cgs/gem/}, version 2.6), MACE \cite{mace} (\url{http://dldcc-web.brc.bcm.edu/lilab/MACE/docs/html/}, version 1.2), peakzilla \cite{peakzilla} (\url{https://github.com/steinmann/peakzilla}).
%, Chung et al., 2013; GEM, Guo et al., 2012; MACE,
%Wang et al., 2014 and Peazilla, Bardet et al., 2013 for the ChIP-exo
%data analysis. 

%For the dPeak algorithm we used the R package
%\textbf{dPeak} version 2.0.1 which is available from
%
%\url{https://github.com/dongjunchung/dpeak}. For the GEM algorithm, we
%used it's Java implementation version 2.6 which is available from
%\url{http://groups.csail.mit.edu/cgs/gem/}. For the Mace algorithm, we
%used it Python implementation version 1.2, which is available from
%\url{http://dldcc-web.brc.bcm.edu/lilab/MACE/docs/html/}. For the
%Peakzilla algorithm, we used the version available in
%\url{https://github.com/steinmann/peakzilla}. 

\SK{Too many repeats of mosaics version etc, try to minimize.}

Candidate regions for
\texttt{dPeak} and \texttt{GEM} were identified for each replicate of
ChIP-exo data using the \textbf{MOSAiCS} algorithm \cite{mosaics} (one
sample analysis using false discovery rate of 1\%) implemented as an \texttt{R}
package \textbf{mosaics} (version 2.9.7 from
Bioconductor). We further filtered out candidate regions by
using the top 400 peaks with highest average ChIP read counts to avoid
potential false positives based on the exploratory analysis. These
regions were also explicitly provided to the GEM algorithm as
candidate regions. Default tuning parameters were used during model
fitting for all methods. Although we were able to download \texttt{CexoR} \cite{cexor} (version 1.8  from \emph{bioconductor}), we were unable to use it for
the $\sig$ experiments because of the \SK{*** issues}.

\subsubsection*{dPeak analysis of $\sig$ ChIP-exo and ChIP-seq data}

We compared the estimated binding events predicted by the
\textbf{MOSAiCS} + \textbf{dPeak} pipeline using reads generated by
ChIP-exo, PE and SE ChIP-seq protocols. We called peaks at a $5\%$ FDR
level using \textbf{MOSAiCS} (the $\mbox{GC content} +
\mbox{Mappability}$ background model for ChIP-exo and the $\mbox{Input only}$
model for PE and SE ChIP-seq). Then, we deconvolved the peaks into
binding events with \textbf{dPeak} (version 2.0.1) by considering a
maximum of 5 binding events within each peak. To avoid false positives, we
only considered ChIP-exo peaks with average ChIP read count greater than
3,000 that overlapped both the SE and PE ChIP-seq peaks. Repeating the analysis with other cutoff values led to similar conclusions.

\subsubsection*{Saturation analysis of ChIP-exo, PE and SE ChIP-seq}
\SK{Next paragraph could use more editing.}
We sub-sampled $N$ reads for both ChIP-exo and SE ChIP-seq
protocols. For PE ChIP-seq, we sub-sampled $N/2$ pairs equaling to a total of $N$ reads. 
For each sub-sample, we identified  peaks using MOSAiCS \cite{mosaics}
($\mbox{GC content} + \mbox{Mappability}$ background model for ChIP-exo and
$\mbox{Input only}$ for SE and PE ChIP-seq) \SK{not sure: for the maximum sample
size}. To avoid potential false positives, we considered only the top 500 peaks
for each data protocol. We defined the number of candidate peaks as
the number of top sample peaks with at least one predicted dPeak binding event; the number of
predicted events is the total number of dPeak predicted binding events; the number of identified $\sig$ sites as the number
of gold-standard $\sig$ sites within 15 bps from an estimated binding
event; and the resolution as the   minimum distance from a
gold-standard $\sig$ site to an estimated binding event. We repeated
this analysis for ten random sub-sampling experiments and reported the median across these experiments.


\SK{Yeah, Materials and Methods has too many repeats, needs to be organized a bit and cleaned up. Figure captions need to be modified and made more easy follow. }
 % Style BST file (bmc-mathphys, vancouver, spbasic).
\bibliographystyle{vancouver}
 \bibliography{chip_exo_paper} 
\nocite{exo_gb}
\nocite{maplot1}
\nocite{maplot2}
\nocite{chipbeyond}
\nocite{meme}




% \nocite{esl}


\newpage

\section{Figures}

\begin{figure}[h!]
  \centering
  \includegraphics[width =
  .95 \textwidth]{figures/fig1/chip_explanation_withEnzyme.pdf}
  % \caption{\textbf{Description of ChIP-exo, SE ChIP-Seq and PE
  %     ChIP-Seq:} A TF is bound to the forward (red) and backward
  %   (blue) strand of the DNA. Then, is sonicated: For ChIP-exo a
  %   exonuclease enzyme (orange hexagon) trims the $5\prime$ ends of
  %   each DNA fragment to a fixed distance from the bound protein,
  %   finally is subjected to Immunoprecipitation and amplification. For
  %   both ChIP-exo and SE ChIP-Seq an adapter is ligated (green
  %   triangles) at the $5\prime$ ends, while for PE ChIP-Seq is ligated
  %   to both ends.}
  \caption{A TF is bound to the forward (red) and backward (blue)
    strand of the DNA. Then, is sonicated: For ChIP-exo a exonuclease
    enzyme (orange hexagon) trims the $5\prime$ ends of each DNA
    fragment to a fixed distance from the bound protein, finally is
    subjected to Immunoprecipitation and amplification. For both
    ChIP-exo and SE ChIP-Seq an adapter is ligated (green triangles)
    at the $5\prime$ ends, while for PE ChIP-Seq is ligated to both
    ends.}
  \label{fig:chip_diagram}
\end{figure}
\newpage

\begin{figure}[h!]
  \centering
  \includegraphics[width = .46\textwidth,page = 3]{figures/fig2/ChIPseqPET_ChIPexo_tagCount_comparison.pdf}
  \includegraphics[width =.46\textwidth]{figures/fig2/forward_strand_ratio_comp_old.pdf}
  \includegraphics[width = .46\textwidth,page = 1]{figures/fig2/eukaryotic_bias_CTCF.pdf}
  \includegraphics[width = .46\textwidth,page = 2]{figures/fig2/eukaryotic_bias_CTCF.pdf}
  \caption{ A) Hexbin plot of PE ChIP-Seq bin counts vs ChIP-exo bin
    counts. B) Forward Strand Ratio densities for SE ChIP-Seq and
    ChIP-exo peaks. C) Mappability score vs mean ChIP tag counts with
    0.95 confidence bands. D) GC - content score vs mean ChIP tag
    counts with 0.95 confidence bands.}
  \label{fig:comp}
\end{figure}

\newpage

\begin{figure}[h!]
\centering
\includegraphics[width = .8 \textwidth]{figures/fig3/scc_ctcf2.pdf}
% \includegraphics[width = .8\textwidth]{/p/keles/ChIPexo/volume3/ChIPexo/figs/for_paper/EColi_strand_cross_corr.pdf}
% \caption{\textbf{SCC curves for $\sig$ samples.} The ``phantom peak''
%   and the summit that corresponds to the read and fragment length
%   respectively are confounded due to the exonuclease digestion.}
\caption{SCC curves for human CTCF on HeLa cell lines.  The SCC curve
  for the ChIP-exo sample from \cite{exo1} is shown in the left panel,
  and the SCC for ChIP-Seq samples from
  \textbf{https://www.encodeproject.org/experiments/ENCSR000AOA/} are
  shown in the right panel. The ChIP-exo curve shows local maxima at
  the motif and read length. Both SE ChIP-Seq curves are maximized at
  the fragments length and show a local maxima at its read length. }
  \label{fig:scc_exo}
\end{figure}

\newpage

\begin{figure}[h!]
  \centering
  \includegraphics[width = \textwidth]{figures/fig4/coverage_diagram3.pdf}
  % \caption{\textbf{Flowchart of the ChIP-exo quality control
  %     pipeline.} A) ARC vs. URCR, this plot presents a global view of
  %   the balance between library complexity and enrichment. There are
  %   two arms: First, one with low ARC, which corresponds to regions
  %   formed by few aligned positions; Second, where the URCR decreases
  %   as the ARC increases. B) Min. depth vs Proportions of regions,
  %   this plot shows the strand composition for all the regions formed
  %   by a min. number of reads. Regions with low depth tend to be
  %   formed by undigested reads from one unique strand, while regions
  %   with higher signal are usually formed by reads in both strands. C)
  %   Min depth vs. FSR, this plot depicts how quickly the FSR's
  %   distribution approximate the median. In a high quality sample, the
  %   median is around 0.5, and the other quantiles reach that value
  %   quickly.}
  \caption{The ChIP-exo reads are partitioned by keeping the regions
    in the genome formed by the undigested fragments. For each region,
    we calculate a collection of summary statistics, which are
    visualized as A) ARC vs. URCR, this plot presents a global view of
    the balance between library complexity and enrichment. There are
    two arms: First, one with low ARC, which corresponds to regions
    formed by few aligned positions; Second, where the URCR decreases
    as the ARC increases. B) Min. depth vs Proportions of regions,
    this plot shows the strand composition for all the regions formed
    by a min. number of reads. Regions with low depth tend to be
    formed by undigested reads from one unique strand, while regions
    with higher signal are usually formed by reads in both strands. C)
    Min depth vs. FSR, this plot depicts how quickly the FSR's
    distribution approximate the median. In a high quality sample, the
    median is around 0.5, and the other quantiles reach that value
    quickly.}
  \label{fig:qcdiagram}
\end{figure}

\newpage

\begin{figure}[h!]
  \centering
  \includegraphics[width = .9  \textwidth,page = 1]{figures/fig5/FoxA1_enrichment.pdf}
  \newline
  \includegraphics[width = .23 \textwidth, page = 1]{figures/fig5/FoxA1_barplots.pdf}
  \includegraphics[width = .23 \textwidth, page = 3]{figures/fig5/FoxA1_barplots.pdf}
  \includegraphics[width =.4\textwidth]{figures/fig5/FoxA1_profiles_around_motif.pdf}
  \includegraphics[width = .3\textwidth,page = 2]{figures/fig5/FoxA1_matched_motif_sequence.pdf}
  \includegraphics[width = .3\textwidth,page = 3]{figures/fig5/FoxA1_matched_motif_sequence.pdf}
  \includegraphics[width = .3\textwidth,page = 1]{figures/fig5/FoxA1_matched_motif_sequence.pdf}
  \caption{Using the mouse FoxA1 experiment from \cite{exoillumina}:
    A) Hexbin plots of $\mbox{ARC}$ against $\mbox{URCR}$, there is a
    slight separation into two strong arms, one corresponds to low
    $\mbox{ARC}$ and varying $\mbox{URCR}$, and for the other
    $\mbox{URCR}$ decreases as $\mbox{ARC}$ increases. B) Number of
    candidate sites for each replicate. C) Percentage of candidate
    sites where the FoxA1 motif was detected. D) Average coverage
    around FoxA1 motif. Base distribution for matched sequence for
    Rep-1 (E), Rep-2 (F) and Rep-3 (G).}
  \label{fig:enrich}
\end{figure}

{\color{red} \emph{need to add labels for each figure and think about
    order, in the supplement we have ecdf from fimo score or pval, and
    overlaps with peaks called by mosaics.
\newline
Also, need to change labels from repk to Rep-k
}}

\newpage

\begin{figure}[h!]
  \centering  
  \includegraphics[width = .9\textwidth,page = 1]{figures/fig6/Strand_imbalance.pdf} 
  \newline
  \includegraphics[width = .9\textwidth,page = 2]{figures/fig6/Strand_imbalance.pdf} 
  \includegraphics[width = .4\textwidth]{figures/fig6/Carroll_FSR_depth_VS_pvalWilcoxon.pdf}
  \caption{Strand imbalance using the mouse FoxA1 experiment from
    \cite{exoillumina}: A) FSR distribution quantiles as the lower
    depth regions are being filtered out, all quantiles approach to
    the median as the lower bound increases. B) Stacked histogram with
    the proportion of regions that are formed by two strands or only
    one, in a good sample the single-stranded regions are going to be
    filtered out quickly as in the middle row. C)
    $-\log_{10}(\text{p.value})$ of testing if the imbalance
    distributions differs when ChIP-exo regions overlap their peaks.}
  \label{fig:strand}
\end{figure}

\newpage

\begin{figure}[h!]
  \centering
  \includegraphics[width = .9  \textwidth,page = 1]{figures/fig7/QC_pipeline_eval_boxplot.pdf}
\newline
  \includegraphics[width = .9  \textwidth,page = 2]{figures/fig7/QC_pipeline_eval_boxplot.pdf}
\newline
  \includegraphics[width = .45  \textwidth,page = 1]{figures/fig7/K562_TBP_sample.pdf}
  \includegraphics[width = .45  \textwidth,page = 2]{figures/fig7/K562_TBP_sample.pdf}
  \caption{Comparison of adjusted Average Read Coverage and
    $\mbox{ARC}$ bias for eukaryotic genome ChIP-exo and ChIP-nexus
    experiment. A) Boxplot of adjusted $\mbox{ARC}$. B) Boxplot of
    large sequencing depth $\mbox{ARC}$'s bias. Comparison of
    sub-sampled ChIP-exo and ChIP-nexus experiments of TBP factor in
    human K562 cell lines: C) Adjusted $\mbox{ARC}$ and D) $\mbox{ARC}$'s
    bias.}
  \label{fig:eval}
\end{figure}

\newpage

\begin{figure}[h!]
  \centering
  \includegraphics[width = .45\textwidth,page = 1]{figures/fig8/methods_comp_FDR5_topM400.pdf}
  \includegraphics[width = .45\textwidth,page = 2]{figures/fig8/methods_comp_FDR5_topM400.pdf}
   \includegraphics[width = .45\textwidth,page = 1]{figures/fig8/sigma_delta_old_densities.pdf}
   \includegraphics[width = .45\textwidth,page = 2]{figures/fig8/sigma_delta_old_densities.pdf}   
  \caption{Comparison of the resolution between dPeak, Gem, Mace and
    Peakzilla methods with \emph{E.Coli} aerobic experiment from the
    first biosample A) Replicate 1 and B) Replicate 2. Resolution is
    defined as the minimum distance between a RegulonDB annotation and
    a predicted binding event. C) $\delta$ parameter in dPeak measures
    average distance of the reads to their respective binding site. In
    ChIP-exo data, reads were located much closer to the binding site
    than in SET ChIP-Seq. D) $\sigma$ parameter measure the dispersion
    of reads around each binding site. In ChIP-exo data, reads showed
    less variation around the their respective binding sites compared
    to SET ChIP-Seq.}
  \label{fig:methods_comp}
\end{figure}

\newpage

\begin{figure}[h]
  \centering
  \includegraphics[width = .9\textwidth]{figures/fig9/saturation_analysis_old.pdf}
  \caption{Comparison of the number of A) candidate regions, B)
    predicted events, C) identified targets and D) resolution among
    ChIP-exo, PE ChIP-Seq and SE ChIP-Seq. RegulonDB annotations are
    considered as a gold standard. A gold standard binding events was
    deemed identified if a binding event was estimated at a $\pm$ 15
    vicinity of it.}
  \label{fig:design}
\end{figure}

\newpage
\begin{figure}[h!]
  \centering
  \includegraphics[width = .46\textwidth]{figures/fig10/sensitivity_exo_olda_data.pdf}
  \includegraphics[width = .46\textwidth]{figures/fig10/resolution_by_dataset_old_data.pdf}
  \caption{Comparison of A) sensitivity and B) resolution between
    ChIP-exo and ChIP-Seq data. Sensitivity is defined as the
    proportion of RegulonDB annotations identified using each
    data. Resolution is defined as the distance between RegulonDB
    annotation and its closest prediction.}
  \label{fig:reso_all}
\end{figure}


\end{document}


% LocalWords:  Ntimes overlaping gaussian repk
